---
output: pdf_document
---




<!DOCTYPE html>
<html lang="en" class=" is-copy-enabled emoji-size-boost is-u2f-enabled">
  <head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# object: http://ogp.me/ns/object# article: http://ogp.me/ns/article# profile: http://ogp.me/ns/profile#">
    <meta charset='utf-8'>

    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/frameworks-d05617edbd0684501ca011b9998eb456541d8156ca96d8ffcfb175827827f62c.css" integrity="sha256-0FYX7b0GhFAcoBG5mY60VlQdgVbKltj/z7F1gngn9iw=" media="all" rel="stylesheet" />
    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/github-a5740f61dda90caedf6fe1f31c4049c578b906cca54de55bb297aff18eb0ff08.css" integrity="sha256-pXQPYd2pDK7fb+HzHEBJxXi5BsylTeVbspev8Y6w/wg=" media="all" rel="stylesheet" />
    
    
    
    

    <link as="script" href="https://assets-cdn.github.com/assets/frameworks-7162beea272a856d06e084945fd8026a47bbb9e7eb295d95b60ea82b26c27296.js" rel="preload" />
    
    <link as="script" href="https://assets-cdn.github.com/assets/github-cb6ab72bc83e90e83a429b044f460572dc5ea6fb1b5894fc185c79e9389986f0.js" rel="preload" />

    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="Content-Language" content="en">
    <meta name="viewport" content="width=device-width">
    
    
    <title>caRpools/CaRpools-extended-PDF.Rmd at master · boutroslab/caRpools</title>
    <link rel="search" type="application/opensearchdescription+xml" href="/opensearch.xml" title="GitHub">
    <link rel="fluid-icon" href="https://github.com/fluidicon.png" title="GitHub">
    <link rel="apple-touch-icon" href="/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">
    <meta property="fb:app_id" content="1401488693436528">

      <meta content="https://avatars2.githubusercontent.com/u/2943832?v=3&amp;s=400" name="twitter:image:src" /><meta content="@github" name="twitter:site" /><meta content="summary" name="twitter:card" /><meta content="boutroslab/caRpools" name="twitter:title" /><meta content="caRpools - Exploratory data analysis for pooled CRISPR/Cas9 screens" name="twitter:description" />
      <meta content="https://avatars2.githubusercontent.com/u/2943832?v=3&amp;s=400" property="og:image" /><meta content="GitHub" property="og:site_name" /><meta content="object" property="og:type" /><meta content="boutroslab/caRpools" property="og:title" /><meta content="https://github.com/boutroslab/caRpools" property="og:url" /><meta content="caRpools - Exploratory data analysis for pooled CRISPR/Cas9 screens" property="og:description" />
      <meta name="browser-stats-url" content="https://api.github.com/_private/browser/stats">
    <meta name="browser-errors-url" content="https://api.github.com/_private/browser/errors">
    <link rel="assets" href="https://assets-cdn.github.com/">
    <link rel="web-socket" href="wss://live.github.com/_sockets/NTgzMjAwNTo3MzQwMzBhNmI0N2RmNzk4MDdiMDAxOTBmYTk3ZjQ5ZDoyY2NiZTE4MjRiMGM5MjA4YTM1NmM2NmE5MDc1ZGUzYTlmZjFiZTczOTA3YWZjYjcyOWM2YzRhNTcyZWQxNDky--77bb6307d4ccfbe424af7d82fa3b4f9f82e2d6e9">
    <meta name="pjax-timeout" content="1000">
    <link rel="sudo-modal" href="/sessions/sudo_modal">

    <meta name="msapplication-TileImage" content="/windows-tile.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="selected-link" value="repo_source" data-pjax-transient>

    <meta name="google-site-verification" content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU">
<meta name="google-site-verification" content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA">
    <meta name="google-analytics" content="UA-3769691-2">

<meta content="collector.githubapp.com" name="octolytics-host" /><meta content="github" name="octolytics-app-id" /><meta content="C13EC2FC:76D2:5460164:575E9308" name="octolytics-dimension-request_id" /><meta content="5832005" name="octolytics-actor-id" /><meta content="francescojm" name="octolytics-actor-login" /><meta content="bc8e5b9cd520d1a54c0e5d02eef1c5b2cb534b36af81565b73407c337a652648" name="octolytics-actor-hash" />
<meta content="/&lt;user-name&gt;/&lt;repo-name&gt;/blob/show" data-pjax-transient="true" name="analytics-location" />



  <meta class="js-ga-set" name="dimension1" content="Logged In">



        <meta name="hostname" content="github.com">
    <meta name="user-login" content="francescojm">

        <meta name="expected-hostname" content="github.com">
      <meta name="js-proxy-site-detection-payload" content="OGQ5MTVkZjA5YjAyMjJkNjg0NzY2M2E1ZTY5YTRlODdmNDA3OWUyNGU2MWJkNWUzNDQyYWRmNTFhZWY3Y2I0YXx7InJlbW90ZV9hZGRyZXNzIjoiMTkzLjYyLjE5NC4yNTIiLCJyZXF1ZXN0X2lkIjoiQzEzRUMyRkM6NzZEMjo1NDYwMTY0OjU3NUU5MzA4IiwidGltZXN0YW1wIjoxNDY1ODE1ODIwfQ==">


      <link rel="mask-icon" href="https://assets-cdn.github.com/pinned-octocat.svg" color="#4078c0">
      <link rel="icon" type="image/x-icon" href="https://assets-cdn.github.com/favicon.ico">

    <meta name="html-safe-nonce" content="bb8dc8b60b7aedf7e2fd339d7e3a906259f78418">
    <meta content="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" name="form-nonce" />

    <meta http-equiv="x-pjax-version" content="46f108ca66e01da98c38cf52e08c76c9">
    

      
  <meta name="description" content="caRpools - Exploratory data analysis for pooled CRISPR/Cas9 screens">
  <meta name="go-import" content="github.com/boutroslab/caRpools git https://github.com/boutroslab/caRpools.git">

  <meta content="2943832" name="octolytics-dimension-user_id" /><meta content="boutroslab" name="octolytics-dimension-user_login" /><meta content="39879518" name="octolytics-dimension-repository_id" /><meta content="boutroslab/caRpools" name="octolytics-dimension-repository_nwo" /><meta content="true" name="octolytics-dimension-repository_public" /><meta content="false" name="octolytics-dimension-repository_is_fork" /><meta content="39879518" name="octolytics-dimension-repository_network_root_id" /><meta content="boutroslab/caRpools" name="octolytics-dimension-repository_network_root_nwo" />
  <link href="https://github.com/boutroslab/caRpools/commits/master.atom" rel="alternate" title="Recent Commits to caRpools:master" type="application/atom+xml">


      <link rel="canonical" href="https://github.com/boutroslab/caRpools/blob/master/CaRpools-extended-PDF.Rmd" data-pjax-transient>
  </head>


  <body class="logged-in   env-production macintosh vis-public page-blob">
    <div id="js-pjax-loader-bar" class="pjax-loader-bar"></div>
    <a href="#start-of-content" tabindex="1" class="accessibility-aid js-skip-to-content">Skip to content</a>

    
    
    



        <div class="header header-logged-in true" role="banner">
  <div class="container clearfix">

    <a class="header-logo-invertocat" href="https://github.com/" data-hotkey="g d" aria-label="Homepage" data-ga-click="Header, go to dashboard, icon:logo">
  <svg aria-hidden="true" class="octicon octicon-mark-github" height="28" version="1.1" viewBox="0 0 16 16" width="28"><path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path></svg>
</a>


        <div class="header-search scoped-search site-scoped-search js-site-search" role="search">
  <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/boutroslab/caRpools/search" class="js-site-search-form" data-scoped-search-url="/boutroslab/caRpools/search" data-unscoped-search-url="/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
    <label class="form-control header-search-wrapper js-chromeless-input-container">
      <div class="header-search-scope">This repository</div>
      <input type="text"
        class="form-control header-search-input js-site-search-focus js-site-search-field is-clearable"
        data-hotkey="s"
        name="q"
        placeholder="Search"
        aria-label="Search this repository"
        data-unscoped-placeholder="Search GitHub"
        data-scoped-placeholder="Search"
        tabindex="1"
        autocapitalize="off">
    </label>
</form></div>


      <ul class="header-nav left" role="navigation">
        <li class="header-nav-item">
          <a href="/pulls" class="js-selected-navigation-item header-nav-link" data-ga-click="Header, click, Nav menu - item:pulls context:user" data-hotkey="g p" data-selected-links="/pulls /pulls/assigned /pulls/mentioned /pulls">
            Pull requests
</a>        </li>
        <li class="header-nav-item">
          <a href="/issues" class="js-selected-navigation-item header-nav-link" data-ga-click="Header, click, Nav menu - item:issues context:user" data-hotkey="g i" data-selected-links="/issues /issues/assigned /issues/mentioned /issues">
            Issues
</a>        </li>
          <li class="header-nav-item">
            <a class="header-nav-link" href="https://gist.github.com/" data-ga-click="Header, go to gist, text:gist">Gist</a>
          </li>
      </ul>

    
<ul class="header-nav user-nav right" id="user-links">
  <li class="header-nav-item">
    
    <a href="/notifications" aria-label="You have unread notifications" class="header-nav-link notification-indicator tooltipped tooltipped-s js-socket-channel js-notification-indicator" data-channel="tenant:1:notification-changed:5832005" data-ga-click="Header, go to notifications, icon:unread" data-hotkey="g n">
        <span class="mail-status unread"></span>
        <svg aria-hidden="true" class="octicon octicon-bell" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M14 12v1H0v-1l.73-.58c.77-.77.81-2.55 1.19-4.42C2.69 3.23 6 2 6 2c0-.55.45-1 1-1s1 .45 1 1c0 0 3.39 1.23 4.16 5 .38 1.88.42 3.66 1.19 4.42l.66.58H14zm-7 4c1.11 0 2-.89 2-2H5c0 1.11.89 2 2 2z"></path></svg>
</a>
  </li>

  <li class="header-nav-item dropdown js-menu-container">
    <a class="header-nav-link tooltipped tooltipped-s js-menu-target" href="/new"
       aria-label="Create new…"
       data-ga-click="Header, create new, icon:add">
      <svg aria-hidden="true" class="octicon octicon-plus left" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 9H7v5H5V9H0V7h5V2h2v5h5z"></path></svg>
      <span class="dropdown-caret"></span>
    </a>

    <div class="dropdown-menu-content js-menu-content">
      <ul class="dropdown-menu dropdown-menu-sw">
        
<a class="dropdown-item" href="/new" data-ga-click="Header, create new repository">
  New repository
</a>

  <a class="dropdown-item" href="/new/import" data-ga-click="Header, import a repository">
    Import repository
  </a>


  <a class="dropdown-item" href="/organizations/new" data-ga-click="Header, create new organization">
    New organization
  </a>



  <div class="dropdown-divider"></div>
  <div class="dropdown-header">
    <span title="boutroslab/caRpools">This repository</span>
  </div>
    <a class="dropdown-item" href="/boutroslab/caRpools/issues/new" data-ga-click="Header, create new issue">
      New issue
    </a>

      </ul>
    </div>
  </li>

  <li class="header-nav-item dropdown js-menu-container">
    <a class="header-nav-link name tooltipped tooltipped-sw js-menu-target" href="/francescojm"
       aria-label="View profile and more"
       data-ga-click="Header, show menu, icon:avatar">
      <img alt="@francescojm" class="avatar" height="20" src="https://avatars2.githubusercontent.com/u/5832005?v=3&amp;s=40" width="20" />
      <span class="dropdown-caret"></span>
    </a>

    <div class="dropdown-menu-content js-menu-content">
      <div class="dropdown-menu dropdown-menu-sw">
        <div class="dropdown-header header-nav-current-user css-truncate">
          Signed in as <strong class="css-truncate-target">francescojm</strong>
        </div>

        <div class="dropdown-divider"></div>

        <a class="dropdown-item" href="/francescojm" data-ga-click="Header, go to profile, text:your profile">
          Your profile
        </a>
        <a class="dropdown-item" href="/stars" data-ga-click="Header, go to starred repos, text:your stars">
          Your stars
        </a>
        <a class="dropdown-item" href="/explore" data-ga-click="Header, go to explore, text:explore">
          Explore
        </a>
          <a class="dropdown-item" href="/integrations" data-ga-click="Header, go to integrations, text:integrations">
            Integrations
          </a>
        <a class="dropdown-item" href="https://help.github.com" data-ga-click="Header, go to help, text:help">
          Help
        </a>


        <div class="dropdown-divider"></div>

        <a class="dropdown-item" href="/settings/profile" data-ga-click="Header, go to settings, icon:settings">
          Settings
        </a>

        <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/logout" class="logout-form" data-form-nonce="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="FPvCyodd9hw/5Q0yOyg4g6avm0/UdO9C0MhxYaYouQ/m+sc0FNwTQLHNEmJnyze3u3jRyvYVwPAUoQXmnwmnzA==" /></div>
          <button class="dropdown-item dropdown-signout" data-ga-click="Header, sign out, icon:logout">
            Sign out
          </button>
</form>      </div>
    </div>
  </li>
</ul>


    
  </div>
</div>


      


    <div id="start-of-content" class="accessibility-aid"></div>

      <div id="js-flash-container">
</div>


    <div role="main" class="main-content">
        <div itemscope itemtype="http://schema.org/SoftwareSourceCode">
    <div id="js-repo-pjax-container" data-pjax-container>
      
<div class="pagehead repohead instapaper_ignore readability-menu experiment-repo-nav">
  <div class="container repohead-details-container">

    

<ul class="pagehead-actions">

  <li>
        <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/notifications/subscribe" class="js-social-container" data-autosubmit="true" data-form-nonce="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="Io50wAxpsgZV0JpC5VPiYR7WaKKfPSKcveq2xol22QiBINh7ujcjot+S3gFLMkk/lXao/e4P744Yc604JD6vBg==" /></div>      <input class="form-control" id="repository_id" name="repository_id" type="hidden" value="39879518" />

        <div class="select-menu js-menu-container js-select-menu">
          <a href="/boutroslab/caRpools/subscription"
            class="btn btn-sm btn-with-count select-menu-button js-menu-target" role="button" tabindex="0" aria-haspopup="true"
            data-ga-click="Repository, click Watch settings, action:blob#show">
            <span class="js-select-button">
              <svg aria-hidden="true" class="octicon octicon-eye" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"></path></svg>
              Watch
            </span>
          </a>
          <a class="social-count js-social-count" href="/boutroslab/caRpools/watchers">
            4
          </a>

        <div class="select-menu-modal-holder">
          <div class="select-menu-modal subscription-menu-modal js-menu-content" aria-hidden="true">
            <div class="select-menu-header js-navigation-enable" tabindex="-1">
              <svg aria-label="Close" class="octicon octicon-x js-menu-close" height="16" role="img" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
              <span class="select-menu-title">Notifications</span>
            </div>

              <div class="select-menu-list js-navigation-container" role="menu">

                <div class="select-menu-item js-navigation-item selected" role="menuitem" tabindex="0">
                  <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
                  <div class="select-menu-item-text">
                    <input checked="checked" id="do_included" name="do" type="radio" value="included" />
                    <span class="select-menu-item-heading">Not watching</span>
                    <span class="description">Be notified when participating or @mentioned.</span>
                    <span class="js-select-button-text hidden-select-button-text">
                      <svg aria-hidden="true" class="octicon octicon-eye" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"></path></svg>
                      Watch
                    </span>
                  </div>
                </div>

                <div class="select-menu-item js-navigation-item " role="menuitem" tabindex="0">
                  <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
                  <div class="select-menu-item-text">
                    <input id="do_subscribed" name="do" type="radio" value="subscribed" />
                    <span class="select-menu-item-heading">Watching</span>
                    <span class="description">Be notified of all conversations.</span>
                    <span class="js-select-button-text hidden-select-button-text">
                      <svg aria-hidden="true" class="octicon octicon-eye" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"></path></svg>
                      Unwatch
                    </span>
                  </div>
                </div>

                <div class="select-menu-item js-navigation-item " role="menuitem" tabindex="0">
                  <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
                  <div class="select-menu-item-text">
                    <input id="do_ignore" name="do" type="radio" value="ignore" />
                    <span class="select-menu-item-heading">Ignoring</span>
                    <span class="description">Never be notified.</span>
                    <span class="js-select-button-text hidden-select-button-text">
                      <svg aria-hidden="true" class="octicon octicon-mute" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8 2.81v10.38c0 .67-.81 1-1.28.53L3 10H1c-.55 0-1-.45-1-1V7c0-.55.45-1 1-1h2l3.72-3.72C7.19 1.81 8 2.14 8 2.81zm7.53 3.22l-1.06-1.06-1.97 1.97-1.97-1.97-1.06 1.06L11.44 8 9.47 9.97l1.06 1.06 1.97-1.97 1.97 1.97 1.06-1.06L13.56 8l1.97-1.97z"></path></svg>
                      Stop ignoring
                    </span>
                  </div>
                </div>

              </div>

            </div>
          </div>
        </div>
</form>
  </li>

  <li>
    
  <div class="js-toggler-container js-social-container starring-container ">

    <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/boutroslab/caRpools/unstar" class="starred" data-form-nonce="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="lYzlPh9r0dc/l+29zJUeQAfOV4FguMRWuUtp8d3bDN2i+JWVLNR/Q6ofqiargxWW4em8MmbdkMM08CsoRTX2bw==" /></div>
      <button
        class="btn btn-sm btn-with-count js-toggler-target"
        aria-label="Unstar this repository" title="Unstar boutroslab/caRpools"
        data-ga-click="Repository, click unstar button, action:blob#show; text:Unstar">
        <svg aria-hidden="true" class="octicon octicon-star" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"></path></svg>
        Unstar
      </button>
        <a class="social-count js-social-count" href="/boutroslab/caRpools/stargazers">
          4
        </a>
</form>
    <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/boutroslab/caRpools/star" class="unstarred" data-form-nonce="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" data-remote="true" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="Z1zGL3JIjrn6dsE43INZprCcv2dr6AT6al7rlUAML4Bxcwc79stdF+RcJBMmAuppMiYbYS8vWdW2hE+m8ZRk8w==" /></div>
      <button
        class="btn btn-sm btn-with-count js-toggler-target"
        aria-label="Star this repository" title="Star boutroslab/caRpools"
        data-ga-click="Repository, click star button, action:blob#show; text:Star">
        <svg aria-hidden="true" class="octicon octicon-star" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"></path></svg>
        Star
      </button>
        <a class="social-count js-social-count" href="/boutroslab/caRpools/stargazers">
          4
        </a>
</form>  </div>

  </li>

  <li>
          <a href="#fork-destination-box" class="btn btn-sm btn-with-count"
              title="Fork your own copy of boutroslab/caRpools to your account"
              aria-label="Fork your own copy of boutroslab/caRpools to your account"
              rel="facebox"
              data-ga-click="Repository, show fork modal, action:blob#show; text:Fork">
              <svg aria-hidden="true" class="octicon octicon-repo-forked" height="16" version="1.1" viewBox="0 0 10 16" width="10"><path d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
            Fork
          </a>

          <div id="fork-destination-box" style="display: none;">
            <h2 class="facebox-header" data-facebox-id="facebox-header">Where should we fork this repository?</h2>
            <include-fragment src=""
                class="js-fork-select-fragment fork-select-fragment"
                data-url="/boutroslab/caRpools/fork?fragment=1">
              <img alt="Loading" height="64" src="https://assets-cdn.github.com/images/spinners/octocat-spinner-128.gif" width="64" />
            </include-fragment>
          </div>

    <a href="/boutroslab/caRpools/network" class="social-count">
      2
    </a>
  </li>
</ul>

    <h1 class="public ">
  <svg aria-hidden="true" class="octicon octicon-repo" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M4 9H3V8h1v1zm0-3H3v1h1V6zm0-2H3v1h1V4zm0-2H3v1h1V2zm8-1v12c0 .55-.45 1-1 1H6v2l-1.5-1.5L3 16v-2H1c-.55 0-1-.45-1-1V1c0-.55.45-1 1-1h10c.55 0 1 .45 1 1zm-1 10H1v2h2v-1h3v1h5v-2zm0-10H2v9h9V1z"></path></svg>
  <span class="author" itemprop="author"><a href="/boutroslab" class="url fn" rel="author">boutroslab</a></span><!--
--><span class="path-divider">/</span><!--
--><strong itemprop="name"><a href="/boutroslab/caRpools" data-pjax="#js-repo-pjax-container">caRpools</a></strong>

</h1>

  </div>
  <div class="container">
    
<nav class="reponav js-repo-nav js-sidenav-container-pjax"
     itemscope
     itemtype="http://schema.org/BreadcrumbList"
     role="navigation"
     data-pjax="#js-repo-pjax-container">

  <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
    <a href="/boutroslab/caRpools" aria-selected="true" class="js-selected-navigation-item selected reponav-item" data-hotkey="g c" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches /boutroslab/caRpools" itemprop="url">
      <svg aria-hidden="true" class="octicon octicon-code" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M9.5 3L8 4.5 11.5 8 8 11.5 9.5 13 14 8 9.5 3zm-5 0L0 8l4.5 5L6 11.5 2.5 8 6 4.5 4.5 3z"></path></svg>
      <span itemprop="name">Code</span>
      <meta itemprop="position" content="1">
</a>  </span>

    <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
      <a href="/boutroslab/caRpools/issues" class="js-selected-navigation-item reponav-item" data-hotkey="g i" data-selected-links="repo_issues repo_labels repo_milestones /boutroslab/caRpools/issues" itemprop="url">
        <svg aria-hidden="true" class="octicon octicon-issue-opened" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
        <span itemprop="name">Issues</span>
        <span class="counter">1</span>
        <meta itemprop="position" content="2">
</a>    </span>

  <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
    <a href="/boutroslab/caRpools/pulls" class="js-selected-navigation-item reponav-item" data-hotkey="g p" data-selected-links="repo_pulls /boutroslab/caRpools/pulls" itemprop="url">
      <svg aria-hidden="true" class="octicon octicon-git-pull-request" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
      <span itemprop="name">Pull requests</span>
      <span class="counter">0</span>
      <meta itemprop="position" content="3">
</a>  </span>

    <a href="/boutroslab/caRpools/wiki" class="js-selected-navigation-item reponav-item" data-hotkey="g w" data-selected-links="repo_wiki /boutroslab/caRpools/wiki">
      <svg aria-hidden="true" class="octicon octicon-book" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M3 5h4v1H3V5zm0 3h4V7H3v1zm0 2h4V9H3v1zm11-5h-4v1h4V5zm0 2h-4v1h4V7zm0 2h-4v1h4V9zm2-6v9c0 .55-.45 1-1 1H9.5l-1 1-1-1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1h5.5l1 1 1-1H15c.55 0 1 .45 1 1zm-8 .5L7.5 3H2v9h6V3.5zm7-.5H9.5l-.5.5V12h6V3z"></path></svg>
      Wiki
</a>

  <a href="/boutroslab/caRpools/pulse" class="js-selected-navigation-item reponav-item" data-selected-links="pulse /boutroslab/caRpools/pulse">
    <svg aria-hidden="true" class="octicon octicon-pulse" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M11.5 8L8.8 5.4 6.6 8.5 5.5 1.6 2.38 8H0v2h3.6l.9-1.8.9 5.4L9 8.5l1.6 1.5H14V8z"></path></svg>
    Pulse
</a>
  <a href="/boutroslab/caRpools/graphs" class="js-selected-navigation-item reponav-item" data-selected-links="repo_graphs repo_contributors /boutroslab/caRpools/graphs">
    <svg aria-hidden="true" class="octicon octicon-graph" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M16 14v1H0V0h1v14h15zM5 13H3V8h2v5zm4 0H7V3h2v10zm4 0h-2V6h2v7z"></path></svg>
    Graphs
</a>

</nav>

  </div>
</div>

<div class="container new-discussion-timeline experiment-repo-nav">
  <div class="repository-content">

    

<a href="/boutroslab/caRpools/blob/233360fcae6857ea544c6537a24416826280f8f8/CaRpools-extended-PDF.Rmd" class="hidden js-permalink-shortcut" data-hotkey="y">Permalink</a>

<!-- blob contrib key: blob_contributors:v21:aa05a35983d3fe313ed7201f0d56b420 -->

<div class="file-navigation js-zeroclipboard-container">
  
<div class="select-menu branch-select-menu js-menu-container js-select-menu left">
  <button class="btn btn-sm select-menu-button js-menu-target css-truncate" data-hotkey="w"
    title="master"
    type="button" aria-label="Switch branches or tags" tabindex="0" aria-haspopup="true">
    <i>Branch:</i>
    <span class="js-select-button css-truncate-target">master</span>
  </button>

  <div class="select-menu-modal-holder js-menu-content js-navigation-container" data-pjax aria-hidden="true">

    <div class="select-menu-modal">
      <div class="select-menu-header">
        <svg aria-label="Close" class="octicon octicon-x js-menu-close" height="16" role="img" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
        <span class="select-menu-title">Switch branches/tags</span>
      </div>

      <div class="select-menu-filters">
        <div class="select-menu-text-filter">
          <input type="text" aria-label="Filter branches/tags" id="context-commitish-filter-field" class="form-control js-filterable-field js-navigation-enable" placeholder="Filter branches/tags">
        </div>
        <div class="select-menu-tabs">
          <ul>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="branches" data-filter-placeholder="Filter branches/tags" class="js-select-menu-tab" role="tab">Branches</a>
            </li>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="tags" data-filter-placeholder="Find a tag…" class="js-select-menu-tab" role="tab">Tags</a>
            </li>
          </ul>
        </div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="branches" role="menu">

        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


            <a class="select-menu-item js-navigation-item js-navigation-open selected"
               href="/boutroslab/caRpools/blob/master/CaRpools-extended-PDF.Rmd"
               data-name="master"
               data-skip-pjax="true"
               rel="nofollow">
              <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
              <span class="select-menu-item-text css-truncate-target js-select-menu-filter-text" title="master">
                master
              </span>
            </a>
        </div>

          <div class="select-menu-no-results">Nothing to show</div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="tags">
        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


            <a class="select-menu-item js-navigation-item js-navigation-open "
              href="/boutroslab/caRpools/tree/v0.83/CaRpools-extended-PDF.Rmd"
              data-name="v0.83"
              data-skip-pjax="true"
              rel="nofollow">
              <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
              <span class="select-menu-item-text css-truncate-target" title="v0.83">
                v0.83
              </span>
            </a>
            <a class="select-menu-item js-navigation-item js-navigation-open "
              href="/boutroslab/caRpools/tree/1.0241-alpha-1/CaRpools-extended-PDF.Rmd"
              data-name="1.0241-alpha-1"
              data-skip-pjax="true"
              rel="nofollow">
              <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
              <span class="select-menu-item-text css-truncate-target" title="1.0241-alpha-1">
                1.0241-alpha-1
              </span>
            </a>
            <a class="select-menu-item js-navigation-item js-navigation-open "
              href="/boutroslab/caRpools/tree/1.025-alpha-1/CaRpools-extended-PDF.Rmd"
              data-name="1.025-alpha-1"
              data-skip-pjax="true"
              rel="nofollow">
              <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"></path></svg>
              <span class="select-menu-item-text css-truncate-target" title="1.025-alpha-1">
                1.025-alpha-1
              </span>
            </a>
        </div>

        <div class="select-menu-no-results">Nothing to show</div>
      </div>

    </div>
  </div>
</div>

  <div class="btn-group right">
    <a href="/boutroslab/caRpools/find/master"
          class="js-pjax-capture-input btn btn-sm"
          data-pjax
          data-hotkey="t">
      Find file
    </a>
    <button aria-label="Copy file path to clipboard" class="js-zeroclipboard btn btn-sm zeroclipboard-button tooltipped tooltipped-s" data-copied-hint="Copied!" type="button">Copy path</button>
  </div>
  <div class="breadcrumb js-zeroclipboard-target">
    <span class="repo-root js-repo-root"><span class="js-path-segment"><a href="/boutroslab/caRpools"><span>caRpools</span></a></span></span><span class="separator">/</span><strong class="final-path">CaRpools-extended-PDF.Rmd</strong>
  </div>
</div>


  <div class="commit-tease">
      <span class="right">
        <a class="commit-tease-sha" href="/boutroslab/caRpools/commit/986a412062197d2354e9419b73a9f6d8e6433b98" data-pjax>
          986a412
        </a>
        <relative-time datetime="2015-10-14T11:48:17Z">Oct 14, 2015</relative-time>
      </span>
      <div>
        <img alt="@jwinter6" class="avatar" height="20" src="https://avatars3.githubusercontent.com/u/7203572?v=3&amp;s=40" width="20" />
        <a href="/jwinter6" class="user-mention" rel="contributor">jwinter6</a>
          <a href="/boutroslab/caRpools/commit/986a412062197d2354e9419b73a9f6d8e6433b98" class="message" data-pjax="true" title="updated Rmd files to include latest changes
added 4 replicate template
updated latest pac page version 0.82.4">updated Rmd files to include latest changes</a>
      </div>

    <div class="commit-tease-contributors">
      <button type="button" class="btn-link muted-link contributors-toggle" data-facebox="#blob_contributors_box">
        <strong>1</strong>
         contributor
      </button>
      
    </div>

    <div id="blob_contributors_box" style="display:none">
      <h2 class="facebox-header" data-facebox-id="facebox-header">Users who have contributed to this file</h2>
      <ul class="facebox-user-list" data-facebox-id="facebox-description">
          <li class="facebox-user-list-item">
            <img alt="@jwinter6" height="24" src="https://avatars1.githubusercontent.com/u/7203572?v=3&amp;s=48" width="24" />
            <a href="/jwinter6">jwinter6</a>
          </li>
      </ul>
    </div>
  </div>

<div class="file">
  <div class="file-header">
  <div class="file-actions">

    <div class="btn-group">
      <a href="/boutroslab/caRpools/raw/master/CaRpools-extended-PDF.Rmd" class="btn btn-sm " id="raw-url">Raw</a>
        <a href="/boutroslab/caRpools/blame/master/CaRpools-extended-PDF.Rmd" class="btn btn-sm js-update-url-with-hash">Blame</a>
      <a href="/boutroslab/caRpools/commits/master/CaRpools-extended-PDF.Rmd" class="btn btn-sm " rel="nofollow">History</a>
    </div>

        <a class="btn-octicon tooltipped tooltipped-nw"
           href="github-mac://openRepo/https://github.com/boutroslab/caRpools?branch=master&amp;filepath=CaRpools-extended-PDF.Rmd"
           aria-label="Open this file in GitHub Desktop"
           data-ga-click="Repository, open with desktop, type:mac">
            <svg aria-hidden="true" class="octicon octicon-device-desktop" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M15 2H1c-.55 0-1 .45-1 1v9c0 .55.45 1 1 1h5.34c-.25.61-.86 1.39-2.34 2h8c-1.48-.61-2.09-1.39-2.34-2H15c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm0 9H1V3h14v8z"></path></svg>
        </a>

        <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/boutroslab/caRpools/edit/master/CaRpools-extended-PDF.Rmd" class="inline-form js-update-url-with-hash" data-form-nonce="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="u6naRmOU9FNgHcAs0R2jIeWr8xfTFBJD4HrDjGliE9PBcGfNfc6cvp+KC1BvQlHL7icxTwR8c2wf0h9xO+64Cw==" /></div>
          <button class="btn-octicon tooltipped tooltipped-nw" type="submit"
            aria-label="Fork this project and edit the file" data-hotkey="e" data-disable-with>
            <svg aria-hidden="true" class="octicon octicon-pencil" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path d="M0 12v3h3l8-8-3-3-8 8zm3 2H1v-2h1v1h1v1zm10.3-9.3L12 6 9 3l1.3-1.3a.996.996 0 0 1 1.41 0l1.59 1.59c.39.39.39 1.02 0 1.41z"></path></svg>
          </button>
</form>        <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="/boutroslab/caRpools/delete/master/CaRpools-extended-PDF.Rmd" class="inline-form" data-form-nonce="ab9ab8d96b0e82079f8c12a4f1e328ef0531a8c2" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="RO9m2Wt2gXEw2CQGqnUyeJMrXCpaa35tVfjoiwDDMazfvUj84RQxCdrXUcGt0NheXyA+XOrlrv5gqh+30h1TeA==" /></div>
          <button class="btn-octicon btn-octicon-danger tooltipped tooltipped-nw" type="submit"
            aria-label="Fork this project and delete the file" data-disable-with>
            <svg aria-hidden="true" class="octicon octicon-trashcan" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M11 2H9c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1H2c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1v9c0 .55.45 1 1 1h7c.55 0 1-.45 1-1V5c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm-1 12H3V5h1v8h1V5h1v8h1V5h1v8h1V5h1v9zm1-10H2V3h9v1z"></path></svg>
          </button>
</form>  </div>

  <div class="file-info">
      1473 lines (1013 sloc)
      <span class="file-info-divider"></span>
    69.9 KB
  </div>
</div>

  
  <div id="readme" class="readme blob instapaper_body">
    <article class="markdown-body entry-content" itemprop="text"><table data-table-type="yaml-metadata">
  <thead>
  <tr>
  <th>title</th>

  <th>output</th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td><div>CaRpools Report - CRISPR Screen Analysis</div></td>

  <td><div><table>
  <thead>
  <tr>
  <th>pdf_document</th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td><div><table>
  <thead>
  <tr>
  <th>fig_height</th>

  <th>fig_width</th>

  <th>highlight</th>

  <th>keep_tex</th>

  <th>latex_engine</th>

  <th>toc</th>

  <th>toc_depth</th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td><div>6</div></td>

  <td><div>11</div></td>

  <td><div></div></td>

  <td><div>true</div></td>

  <td><div>xelatex</div></td>

  <td><div>true</div></td>

  <td><div>3</div></td>
  </tr>
  </tbody>
</table></div></td>
  </tr>
  </tbody>
</table></div></td>
  </tr>
  </tbody>
</table>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(<span class="pl-s"><span class="pl-pds">"</span>caRpools<span class="pl-pds">"</span></span> <span class="pl-k">%in%</span> rownames(installed.packages()) <span class="pl-k">==</span> <span class="pl-c1">FALSE</span>) {install.packages(<span class="pl-s"><span class="pl-pds">"</span>caRpools<span class="pl-pds">"</span></span>)}
library(<span class="pl-smi">caRpools</span>,<span class="pl-v">warn.conflicts</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">quietly</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,<span class="pl-v">verbose</span> <span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
load.packages()</pre></div>

<p>\newpage</p>

<p>\begin{center}
\includegraphics{CaRpools.png}
\end{center}</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#OPTIONAL PROXY</span>
<span class="pl-c">#options(RCurlOptions=list(proxy="YOURproxyHERE", http.version=1))</span>


<span class="pl-c">### Load PARAMETER FILE</span>
<span class="pl-c"># This name can be changed to your needs</span>
<span class="pl-k">if</span>(<span class="pl-k">!</span>exists(<span class="pl-s"><span class="pl-pds">"</span>miaccs.file<span class="pl-pds">"</span></span>))
  {<span class="pl-v">miaccs.file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>MIACCS.xls<span class="pl-pds">"</span></span>}
<span class="pl-v">miaccs</span> <span class="pl-k">=</span> load.file(<span class="pl-smi">miaccs.file</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>xlsx<span class="pl-pds">"</span></span>)

<span class="pl-c">####  global options to produce the report</span>
<span class="pl-e">knitr</span><span class="pl-k">::</span><span class="pl-smi">opts_chunk</span><span class="pl-k">$</span>set(<span class="pl-v">sanitize</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-e">knitr</span><span class="pl-k">::</span><span class="pl-smi">opts_chunk</span><span class="pl-k">$</span>set(<span class="pl-v">dev</span><span class="pl-k">=</span><span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.device<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>]) <span class="pl-c"># ImageMagick needs to be installed!</span>
<span class="pl-e">knitr</span><span class="pl-k">::</span><span class="pl-smi">opts_chunk</span><span class="pl-k">$</span>set(<span class="pl-v">dpi</span><span class="pl-k">=</span>as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.dpi<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>]))
<span class="pl-e">knitr</span><span class="pl-k">::</span><span class="pl-smi">opts_chunk</span><span class="pl-k">$</span>set(<span class="pl-v">results</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>asis<span class="pl-pds">"</span></span>)
options(<span class="pl-v">scipen</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>)

<span class="pl-c">####  Screen Information</span>
<span class="pl-v">screen.id</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>screening.id<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.date</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>screening.date<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.hypothesis</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>hypothesis<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.description</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.description<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.organism</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.organism<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.cell</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.cellline<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.drug</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.treated<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.control</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.untreated<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.targets</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.genes<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.plasmids</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.plasmids<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.library</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.library<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.NGS</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>lib.sequencing<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">screen.despergene</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.designs<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">####  FILES AND DATASETS</span>

<span class="pl-c">##  PATHs</span>
<span class="pl-c"># abolsute path to where CRISPR-extract.pl and CRISPR-mapping.pl are located</span>
<span class="pl-v">scriptpath</span><span class="pl-k">=</span>as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.scriptpath<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># abolsute path to where fastq / read count and mapping files are located</span>
<span class="pl-v">datapath</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.datapath<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">##  File Names</span>

<span class="pl-c"># name of fastq files of the datasets WITHOUT extension .fastq</span>
<span class="pl-c"># in general, the presence of two replicates for each group is MANDATORY.</span>
<span class="pl-c"># If mapping = TRUE and extract = FALSE, these files need to be already extracted fastq files.</span>
<span class="pl-c"># If mapping = FALSE and extract = FALSE, these files need to be final .txt files with read count data and .txt extension.</span>

<span class="pl-v">fileCONTROL1</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.untreated1<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">d.CONTROL1</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.untreated1.desc<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">fileCONTROL2</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.untreated2<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">d.CONTROL2</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.untreated2.desc<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">fileTREAT1</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.treated1<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">d.TREAT1</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.treated1.desc<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">fileTREAT2</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.treated2<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">d.TREAT2</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.treated2.desc<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># name of reference library .fasta file WITHOUT extension .fasta</span>
<span class="pl-v">referencefile</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.lib.ref<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">##  Analysis Name</span>
<span class="pl-c"># This name will be used as file name for all tabular, PDF and HTML files.</span>
<span class="pl-v">analysis.name</span> <span class="pl-k">=</span> <span class="pl-e">stringi</span><span class="pl-k">::</span>stri_replace_all_fixed(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.name<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>], <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>)
<span class="pl-e">knitr</span><span class="pl-k">::</span><span class="pl-smi">opts_chunk</span><span class="pl-k">$</span>set(<span class="pl-v">fig.path</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>,<span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>,<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>))


<span class="pl-c">####  PERL REGULAR EXPRESSIONS</span>

<span class="pl-c">##  Gene Identifier Discrimination from sgRNA name</span>
<span class="pl-c"># Regular expression that is returning the gene identifier from the sgRNA name</span>
<span class="pl-c"># e.g. expression("^(.+?)(_.+)")</span>
<span class="pl-v">g.extractpattern</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.regex.gene<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Pattern to extract data if extract=TRUE, for more information see manual</span>
<span class="pl-v">seq.pattern</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.regex.extract<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Pattern of READ ID (depends on sequencing maschine), e.g. M01100</span>
<span class="pl-v">maschine.pattern</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.regex.maschine<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])


<span class="pl-c">####  DATA EXTRACTION AND MAPPING</span>

<span class="pl-c">## Extract fastq files from fastq raw data?</span>
<span class="pl-v">extract</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.extract.fastq<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-c"># is the data within the fastq file in reverse complement oriantation (e.g. paired end read)</span>
<span class="pl-v">reversecomplement</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.fastq.rev<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])


<span class="pl-c">## Mapping</span>
<span class="pl-c"># Shall the extracted fastq file be mapped to the reference library file using the generated or already present .SAM file and count files created?</span>
<span class="pl-v">mapping</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.mapping<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">## Mapping extracted data</span>
<span class="pl-c"># Create a bowtie2 index file? TRUE if NO bowtie2 index file has been generated for the reference library fasta file, so that a new one is created</span>
<span class="pl-v">createindex</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bt2.index<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])


<span class="pl-c"># How many threads shall be used for bowtie2 usage?</span>
<span class="pl-v">threads</span> <span class="pl-k">=</span> as.integer(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bt2.threads<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-c"># Additional parameters for bowtie2 alignment</span>
<span class="pl-v">bowtieparams</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bt2.additional<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-k">if</span>(is.na(<span class="pl-smi">bowtieparams</span>) <span class="pl-k">||</span> <span class="pl-smi">bowtieparams</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>NA<span class="pl-pds">"</span></span> <span class="pl-k">||</span> <span class="pl-smi">bowtieparams</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span> <span class="pl-k">||</span> is.null(<span class="pl-smi">bowtieparams</span>))
  {<span class="pl-v">bowtieparams</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>}
<span class="pl-c"># Sensitivity string for bowtie2 alignment, e.g. --very-sensitive-local, --local, --very-sensitive -&gt; have a look in the bowtie2 manual!</span>
<span class="pl-v">sensitivity</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bt2.sens<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-c"># How good must the alignment of the 20 nt CRISPR Oligo must be in order to be considered as present in the data?</span>
<span class="pl-v">match</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.align.quality<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])


<span class="pl-c">####  REPORT GENERATION</span>

<span class="pl-c">## CONVERT GENE IDENTIFIER?</span>
<span class="pl-v">g.convert</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.gconvert<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Which dataset shall be used?</span>
<span class="pl-c"># See biomaRt for more information on dataset and type</span>
<span class="pl-v">a.database</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bm.database<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">a.dataset</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bm.dataset<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])


<span class="pl-c"># What kind of identifier is used? e.g. EnsemblID or HGNC Symbol?</span>
<span class="pl-c"># See biomaRt Filter for additional identifiers (filters)</span>
<span class="pl-v">g.identifier</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bm.identifier<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Convert it to?</span>
<span class="pl-c"># see biomaRt for additional attributes</span>
<span class="pl-v">g.identifier.new</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bm.identifier.new<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">## Anootate final hit candidates?</span>
<span class="pl-v">a.annotate.hits</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.annotate<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-v">a.annotate</span> <span class="pl-k">=</span> unlist(strsplit(<span class="pl-e">stringi</span><span class="pl-k">::</span>stri_replace_all_fixed(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.bm.attributes<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>],<span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">fixed</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>), <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>, <span class="pl-v">fixed</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>))


<span class="pl-c">####  GENERAL</span>

<span class="pl-c"># Add statistics to plots?</span>
<span class="pl-v">plot.statistic</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.statistic<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Column in which the gene/sgRNA identifier is</span>
<span class="pl-v">namecolumn</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.namecolumn<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Column in which the read count data is</span>
<span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.readcolumn<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Normalize read count of datasets in the plots?</span>
<span class="pl-v">normalize</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.normalize<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Normalization function used to normalize datasets</span>
<span class="pl-v">norm.function</span> <span class="pl-k">=</span> eval(parse(<span class="pl-v">text</span> <span class="pl-k">=</span> <span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.normfun<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>]))

<span class="pl-c"># color to plot highlighted genes / sgRNAs</span>
<span class="pl-v">plot.labelcolor</span> <span class="pl-k">=</span> as.character(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.labelcolor<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-v">put.names</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.labelnames<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">#### DATA ANALYSIS</span>
<span class="pl-c"># Targeting Controls</span>
<span class="pl-v">controls.target</span> <span class="pl-k">=</span> as.character(unlist(strsplit(<span class="pl-e">stringi</span><span class="pl-k">::</span>stri_replace_all_fixed(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.pos<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>],<span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">fixed</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>), <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>, <span class="pl-v">fixed</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)))
<span class="pl-k">if</span>(is.na(<span class="pl-smi">controls.target</span>) <span class="pl-k">||</span> <span class="pl-smi">controls.target</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>NULL<span class="pl-pds">"</span></span> <span class="pl-k">||</span> <span class="pl-smi">controls.target</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>)
  {
  <span class="pl-v">controls.target</span> <span class="pl-k">=</span> <span class="pl-c1">NULL</span>
  }

<span class="pl-c"># Non-Targeting Controls</span>
<span class="pl-v">controls.nontarget</span> <span class="pl-k">=</span> as.character(unlist(strsplit(<span class="pl-e">stringi</span><span class="pl-k">::</span>stri_replace_all_fixed(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.non-targeting<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>],<span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, <span class="pl-v">fixed</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>), <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>, <span class="pl-v">fixed</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)))
<span class="pl-k">if</span>(is.na(<span class="pl-smi">controls.nontarget</span>) <span class="pl-k">||</span> <span class="pl-smi">controls.nontarget</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>NULL<span class="pl-pds">"</span></span> <span class="pl-k">||</span> <span class="pl-smi">controls.nontarget</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span> <span class="pl-k">||</span> <span class="pl-smi">controls.nontarget</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  {
  <span class="pl-v">controls.nontarget</span> <span class="pl-k">=</span> <span class="pl-c1">NULL</span>
  }

<span class="pl-c">##  wilcoxn</span>
<span class="pl-v">control.picks</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.randompicks<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">##  CUT OFF</span>
<span class="pl-v">sig.pval.deseq</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.deseq2<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">sig.pval.mageck</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.mageck<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">sig.pval.wilcox</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.wilcox<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c">## HIT VISUALIZATION</span>
<span class="pl-c"># This will plot FOUR scatter plots for each enriched or depleted hit automatically and label the genes and sgRNAs</span>

<span class="pl-v">sgRNAs.top.deseq</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.topdeseq<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">sgRNAs.top.mageck</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.topmageck<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">cutoff.override</span> <span class="pl-k">=</span> as.logical(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.override<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Percentage of TOP enriched or depleted genes to plot in general</span>
<span class="pl-v">number.hits.plot.enriched</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.top.enriched<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
<span class="pl-v">number.hits.plot.depleted</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.top.depleted<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])

<span class="pl-c"># Comparing Methods</span>
<span class="pl-v">compare.cutoff</span> <span class="pl-k">=</span> as.numeric(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>carpools.overrride.genes<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>])
</pre></div>

<p>\newpage</p>

<h1><a id="user-content-screen" class="anchor" href="#screen" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Screen</h1>

<table><thead>
<tr>
<th>Screen</th>
<th>Information</th>
</tr>
</thead><tbody>
<tr>
<td><strong>Screen ID</strong></td>
<td><code>r screen.id</code></td>
</tr>
<tr>
<td><strong>Screening Date</strong></td>
<td><code>r screen.date</code></td>
</tr>
<tr>
<td><strong>Organism</strong></td>
<td><code>r screen.organism</code></td>
</tr>
<tr>
<td><strong>Cell Line</strong></td>
<td><code>r screen.cell</code></td>
</tr>
<tr>
<td><strong>Drug</strong></td>
<td><code>r screen.drug</code></td>
</tr>
<tr>
<td><strong>Control</strong></td>
<td><code>r screen.control</code></td>
</tr>
<tr>
<td><strong>Number of Target</strong></td>
<td><code>r screen.targets</code></td>
</tr>
<tr>
<td><strong>Designs per Gene</strong></td>
<td><code>r screen.despergene</code></td>
</tr>
<tr>
<td><strong>Library</strong></td>
<td><code>r screen.library</code></td>
</tr>
<tr>
<td><strong>Library Reference File</strong></td>
<td><code>r referencefile</code></td>
</tr>
<tr>
<td><strong>NGS</strong></td>
<td><code>r screen.NGS</code></td>
</tr>
</tbody></table>

<p><strong>MIACCS</strong><br>
The MIACCS-file can be found at  </p>

<p><code>r paste("_", datapath, "/", "_" , sep="")</code>  </p>

<p><code>r paste("__", "MIACCS.xls" ,"__", sep="")</code></p>

<p><strong>Description</strong><br>
<code>r screen.description</code>  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-c">## Prepare for plotting overviews</span>
<span class="pl-k">if</span>(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.plasmids.image<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>] <span class="pl-k">!=</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span> <span class="pl-k">&amp;&amp;</span> <span class="pl-k">!</span>is.na(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.plasmids.image<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>]))
  {
    <span class="pl-v">plasmid.image</span> <span class="pl-k">=</span> strsplit(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>assay.plasmids.image<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>], <span class="pl-v">split</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
    <span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(unlist(<span class="pl-smi">plasmid.image</span>)))
      {
        cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span>#<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Plasmid Overview<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span>![](<span class="pl-pds">"</span></span>,paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">plasmid.image</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">i</span>], <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-s"><span class="pl-pds">"</span>)<span class="pl-pds">"</span></span> )
      }

  }

<span class="pl-k">if</span>(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>cell.setup.image<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>] <span class="pl-k">!=</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span><span class="pl-k">&amp;&amp;</span> <span class="pl-k">!</span>is.na(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>cell.setup.image<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>]))
  {
    <span class="pl-v">screen.image</span> <span class="pl-k">=</span> strsplit(<span class="pl-smi">miaccs</span>[<span class="pl-s"><span class="pl-pds">"</span>cell.setup.image<span class="pl-pds">"</span></span>,<span class="pl-c1">3</span>], <span class="pl-v">split</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)
    <span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(unlist(<span class="pl-smi">screen.image</span>)))
      {
        cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span>#<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Experimental Setup<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
        cat(<span class="pl-s"><span class="pl-pds">"</span>![](<span class="pl-pds">"</span></span>,paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">screen.image</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">i</span>], <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>),<span class="pl-s"><span class="pl-pds">"</span>)<span class="pl-pds">"</span></span> )
      }
  }
</pre></div>

<p>\newpage</p>

<h1><a id="user-content-load-data" class="anchor" href="#load-data" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Load Data</h1>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Do we need to map the data?</span>
<span class="pl-k">if</span>(identical(<span class="pl-smi">mapping</span>, <span class="pl-c1">TRUE</span>))
  {
    <span class="pl-c"># Extract Data?</span>
   <span class="pl-v">fileCONTROL1</span> <span class="pl-k">=</span> data.extract(<span class="pl-smi">scriptpath</span>, <span class="pl-smi">datapath</span>, <span class="pl-v">fastqfile</span><span class="pl-k">=</span><span class="pl-smi">fileCONTROL1</span>, <span class="pl-smi">extract</span>, <span class="pl-smi">seq.pattern</span>, <span class="pl-smi">maschine.pattern</span>, <span class="pl-smi">createindex</span>, <span class="pl-smi">referencefile</span>, <span class="pl-smi">mapping</span>, <span class="pl-smi">reversecomplement</span>, <span class="pl-smi">threads</span>, <span class="pl-smi">bowtieparams</span>, <span class="pl-smi">sensitivity</span>, <span class="pl-smi">match</span>)

   <span class="pl-v">fileCONTROL2</span> <span class="pl-k">=</span> data.extract(<span class="pl-smi">scriptpath</span>, <span class="pl-smi">datapath</span>, <span class="pl-v">fastqfile</span><span class="pl-k">=</span><span class="pl-smi">fileCONTROL2</span>, <span class="pl-smi">extract</span>, <span class="pl-smi">seq.pattern</span>, <span class="pl-smi">maschine.pattern</span>, <span class="pl-smi">createindex</span>, <span class="pl-smi">referencefile</span>, <span class="pl-smi">mapping</span>, <span class="pl-smi">reversecomplement</span>, <span class="pl-smi">threads</span>, <span class="pl-smi">bowtieparams</span>, <span class="pl-smi">sensitivity</span>, <span class="pl-smi">match</span>)

   <span class="pl-v">fileTREAT1</span> <span class="pl-k">=</span> data.extract(<span class="pl-smi">scriptpath</span>, <span class="pl-smi">datapath</span>, <span class="pl-v">fastqfile</span><span class="pl-k">=</span><span class="pl-smi">fileTREAT1</span>, <span class="pl-smi">extract</span>, <span class="pl-smi">seq.pattern</span>, <span class="pl-smi">maschine.pattern</span>, <span class="pl-smi">createindex</span>, <span class="pl-smi">referencefile</span>, <span class="pl-smi">mapping</span>, <span class="pl-smi">reversecomplement</span>, <span class="pl-smi">threads</span>, <span class="pl-smi">bowtieparams</span>, <span class="pl-smi">sensitivity</span>, <span class="pl-smi">match</span>)

   <span class="pl-v">fileTREAT2</span> <span class="pl-k">=</span> data.extract(<span class="pl-smi">scriptpath</span>, <span class="pl-smi">datapath</span>, <span class="pl-v">fastqfile</span><span class="pl-k">=</span><span class="pl-smi">fileTREAT2</span>, <span class="pl-smi">extract</span>, <span class="pl-smi">seq.pattern</span>, <span class="pl-smi">maschine.pattern</span>, <span class="pl-smi">createindex</span>, <span class="pl-smi">referencefile</span>, <span class="pl-smi">mapping</span>, <span class="pl-smi">reversecomplement</span>, <span class="pl-smi">threads</span>, <span class="pl-smi">bowtieparams</span>, <span class="pl-smi">sensitivity</span>, <span class="pl-smi">match</span>)



  <span class="pl-c"># Data will be mapped, so we start mapping and then get back the data!</span>
    <span class="pl-v">CONTROL1</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileCONTROL1</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">CONTROL2</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileCONTROL2</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">TREAT1</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileTREAT1</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">TREAT2</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileTREAT2</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">libFILE</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">referencefile</span>,<span class="pl-s"><span class="pl-pds">"</span>.fasta<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>),<span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>fastalib<span class="pl-pds">"</span></span>)
 } <span class="pl-k">else</span> {
    <span class="pl-c"># Extracted and mapped read count data is present, so ONLY load files already there using the filename fileCONTROL etc.</span>
    <span class="pl-v">CONTROL1</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileCONTROL1</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">CONTROL2</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileCONTROL2</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">TREAT1</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileTREAT1</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">TREAT2</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, <span class="pl-smi">fileTREAT2</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>))
    <span class="pl-v">libFILE</span> <span class="pl-k">=</span> load.file(paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">referencefile</span>,<span class="pl-s"><span class="pl-pds">"</span>.fasta<span class="pl-pds">"</span></span>,<span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>),<span class="pl-v">header</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>fastalib<span class="pl-pds">"</span></span>)
}
</pre></div>

<p><strong>A brief look into the data passed on to the analysis scripts:</strong>  </p>

<p><code>r # make randoms for brief stats table</code>
<code>r sample.sgrna = sample(1:nrow(CONTROL1),10)</code></p>

<p><strong><code>r d.CONTROL1</code></strong><br>
<strong><code>r fileCONTROL1</code></strong>
<code>r knitr::kable(CONTROL1[sample.sgrna,])</code></p>

<p><strong><code>r d.CONTROL2</code></strong><br>
<strong><code>r fileCONTROL2</code></strong>
<code>r knitr::kable(CONTROL2[sample.sgrna,])</code></p>

<p><strong><code>r d.TREAT1</code></strong><br>
<strong><code>r fileTREAT1</code></strong>
<code>r knitr::kable(TREAT1[sample.sgrna,])</code></p>

<p><strong><code>r d.TREAT2</code></strong><br>
<strong><code>r fileTREAT2</code></strong>
<code>r knitr::kable(TREAT2[sample.sgrna,])</code></p>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(identical(<span class="pl-smi">g.convert</span>, <span class="pl-c1">TRUE</span>))
{
  cat(<span class="pl-s"><span class="pl-pds">"</span>## Convert Gene Identifiers<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

  cat(<span class="pl-s"><span class="pl-pds">"</span>Gene identifiers are converted from<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>__<span class="pl-pds">"</span></span>, <span class="pl-smi">g.identifier</span>, <span class="pl-s"><span class="pl-pds">"</span>__<span class="pl-pds">"</span></span> ,<span class="pl-s"><span class="pl-pds">"</span>to __<span class="pl-pds">"</span></span>,<span class="pl-smi">g.identifier.new</span>,<span class="pl-s"><span class="pl-pds">"</span>__<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>using the __<span class="pl-pds">"</span></span>, <span class="pl-smi">a.dataset</span>,<span class="pl-s"><span class="pl-pds">"</span>__ database.<span class="pl-pds">"</span></span> ,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    <span class="pl-c"># convert from ENSEMBL ID to human genome nomenclature symbol</span>
    <span class="pl-v">CONTROL1</span> <span class="pl-k">=</span> get.gene.info(<span class="pl-smi">CONTROL1</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-smi">g.identifier.new</span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

    <span class="pl-v">CONTROL2</span> <span class="pl-k">=</span> get.gene.info(<span class="pl-smi">CONTROL2</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-smi">g.identifier.new</span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

    <span class="pl-v">TREAT1</span> <span class="pl-k">=</span> get.gene.info(<span class="pl-smi">TREAT1</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-smi">g.identifier.new</span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

    <span class="pl-v">TREAT2</span> <span class="pl-k">=</span> get.gene.info(<span class="pl-smi">TREAT2</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-smi">g.identifier.new</span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

    <span class="pl-c"># convert control list</span>
     <span class="pl-v">controls.target</span> <span class="pl-k">=</span> get.gene.info(as.data.frame(<span class="pl-smi">controls.target</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-smi">g.identifier.new</span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>, <span class="pl-v">controls</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
    <span class="pl-v">controls.target</span> <span class="pl-k">=</span> <span class="pl-smi">controls.target</span>[,<span class="pl-c1">1</span>]

    <span class="pl-c"># Library file</span>
    <span class="pl-v">libFILE</span> <span class="pl-k">=</span> get.gene.info(<span class="pl-smi">libFILE</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-smi">g.identifier.new</span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)
}

<span class="pl-c"># aggregate to genes for later plots</span>
<span class="pl-v">CONTROL1.g</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">CONTROL1</span>, <span class="pl-v">agg.function</span><span class="pl-k">=</span><span class="pl-smi">sum</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">CONTROL2.g</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">CONTROL2</span>, <span class="pl-v">agg.function</span><span class="pl-k">=</span><span class="pl-smi">sum</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">TREAT1.g</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">TREAT1</span>, <span class="pl-v">agg.function</span><span class="pl-k">=</span><span class="pl-smi">sum</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">TREAT2.g</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">TREAT2</span>, <span class="pl-v">agg.function</span><span class="pl-k">=</span><span class="pl-smi">sum</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>)

<span class="pl-c"># Add genes names to normal count file (if not present)</span>
<span class="pl-v">CONTROL1</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">CONTROL1</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>annotate<span class="pl-pds">"</span></span>)
<span class="pl-v">CONTROL2</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">CONTROL2</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>annotate<span class="pl-pds">"</span></span>)
<span class="pl-v">TREAT1</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">TREAT1</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>,<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>annotate<span class="pl-pds">"</span></span>)
<span class="pl-v">TREAT2</span><span class="pl-k">=</span>aggregatetogenes(<span class="pl-smi">TREAT2</span>, <span class="pl-v">extractpattern</span> <span class="pl-k">=</span> <span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>annotate<span class="pl-pds">"</span></span>)

<span class="pl-c"># Calculate Percentage into gene rows</span>
<span class="pl-v">number.enriched.counter</span> <span class="pl-k">=</span> ceiling((nrow(<span class="pl-smi">CONTROL1.g</span>)<span class="pl-k">/</span><span class="pl-c1">100</span>)<span class="pl-k">*</span><span class="pl-smi">number.hits.plot.enriched</span>)
<span class="pl-v">number.depleted.counter</span> <span class="pl-k">=</span> ceiling((nrow(<span class="pl-smi">CONTROL1.g</span>)<span class="pl-k">/</span><span class="pl-c1">100</span>)<span class="pl-k">*</span><span class="pl-smi">number.hits.plot.depleted</span>)
</pre></div>

<p>\newpage</p>

<h1><a id="user-content-stats" class="anchor" href="#stats" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Stats</h1>

<p>All file-based output (e.g. tables) is stored in:<br>
<em><code>r datapath</code></em>.</p>

<h2><a id="user-content-general" class="anchor" href="#general" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>General</h2>

<p>General stats can be found in<br>
<strong><code>r paste(analysis.name, "STATS.xls", sep="-")</code></strong>.<br>
The following read count statistics were calculated for the single datasets.  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># General</span>
<span class="pl-v">U1.stats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">CONTROL1</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>stats<span class="pl-pds">"</span></span>)
<span class="pl-v">U2.stats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">CONTROL2</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>stats<span class="pl-pds">"</span></span>)
<span class="pl-v">T1.stats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">TREAT1</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>stats<span class="pl-pds">"</span></span>)
<span class="pl-v">T2.stats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">TREAT2</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>stats<span class="pl-pds">"</span></span>)

<span class="pl-v">combined.stats</span> <span class="pl-k">=</span> cbind.data.frame(<span class="pl-smi">U1.stats</span>[,<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">2</span>], <span class="pl-smi">U2.stats</span>[,<span class="pl-c1">2</span>], <span class="pl-smi">T1.stats</span>[,<span class="pl-c1">2</span>], <span class="pl-smi">T2.stats</span>[,<span class="pl-c1">2</span>])
colnames(<span class="pl-smi">combined.stats</span>) <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Readcount<span class="pl-pds">"</span></span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>)

<span class="pl-c"># output to report</span>
<span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">combined.stats</span>)

<span class="pl-c"># output to file</span>
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">combined.stats</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>STATS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Combined Stats<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)</pre></div>

<p>\newpage</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># General</span>
<span class="pl-v">U1.unmapped</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">CONTROL1</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">U1.unmapped</span> <span class="pl-k">=</span> <span class="pl-smi">U1.unmapped</span>[order(<span class="pl-smi">U1.unmapped</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]

<span class="pl-v">U2.unmapped</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">CONTROL2</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">U2.unmapped</span> <span class="pl-k">=</span> <span class="pl-smi">U2.unmapped</span>[order(<span class="pl-smi">U2.unmapped</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]

<span class="pl-v">T1.unmapped</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">TREAT1</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">T1.unmapped</span> <span class="pl-k">=</span> <span class="pl-smi">T1.unmapped</span>[order(<span class="pl-smi">T1.unmapped</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]

<span class="pl-v">T2.unmapped</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">TREAT2</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">T2.unmapped</span> <span class="pl-k">=</span> <span class="pl-smi">T2.unmapped</span>[order(<span class="pl-smi">T2.unmapped</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]

<span class="pl-c"># output to file</span>
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">U1.unmapped</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-smi">d.CONTROL1</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">U2.unmapped</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-smi">d.CONTROL2</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">T1.unmapped</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-smi">d.TREAT1</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">T2.unmapped</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-smi">d.TREAT2</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

<span class="pl-c"># Output single sgRNAs of each gene</span>
<span class="pl-v">U1.unmapped.singles</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">CONTROL1</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-smi">U1.unmapped</span>[,<span class="pl-s"><span class="pl-pds">"</span>name<span class="pl-pds">"</span></span>], <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">U1.unmapped.singles</span> <span class="pl-k">=</span> <span class="pl-smi">U1.unmapped.singles</span>[order(<span class="pl-smi">U1.unmapped.singles</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">U1.unmapped.singles</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span>paste(<span class="pl-smi">d.CONTROL1</span>,<span class="pl-s"><span class="pl-pds">"</span>sgRNA<span class="pl-pds">"</span></span>), <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-v">U2.unmapped.singles</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">CONTROL2</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-smi">U2.unmapped</span>[,<span class="pl-s"><span class="pl-pds">"</span>name<span class="pl-pds">"</span></span>], <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">U2.unmapped.singles</span> <span class="pl-k">=</span> <span class="pl-smi">U2.unmapped.singles</span>[order(<span class="pl-smi">U2.unmapped.singles</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">U2.unmapped.singles</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span>paste(<span class="pl-smi">d.CONTROL2</span>,<span class="pl-s"><span class="pl-pds">"</span>sgRNA<span class="pl-pds">"</span></span>), <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-v">T1.unmapped.singles</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">TREAT1</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-smi">T1.unmapped</span>[,<span class="pl-s"><span class="pl-pds">"</span>name<span class="pl-pds">"</span></span>], <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">T1.unmapped.singles</span> <span class="pl-k">=</span> <span class="pl-smi">T1.unmapped.singles</span>[order(<span class="pl-smi">T1.unmapped.singles</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">T1.unmapped.singles</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span>paste(<span class="pl-smi">d.TREAT1</span>,<span class="pl-s"><span class="pl-pds">"</span>sgRNA<span class="pl-pds">"</span></span>), <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-v">T2.unmapped.singles</span> <span class="pl-k">=</span> unmapped.genes(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">TREAT2</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">genes</span><span class="pl-k">=</span><span class="pl-smi">T2.unmapped</span>[,<span class="pl-s"><span class="pl-pds">"</span>name<span class="pl-pds">"</span></span>], <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>)
<span class="pl-v">T2.unmapped.singles</span> <span class="pl-k">=</span> <span class="pl-smi">T2.unmapped.singles</span>[order(<span class="pl-smi">T2.unmapped.singles</span><span class="pl-k">$</span><span class="pl-smi">sgRNA</span>, <span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">T2.unmapped.singles</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DROPOUT.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span>paste(<span class="pl-smi">d.TREAT2</span>,<span class="pl-s"><span class="pl-pds">"</span>sgRNA<span class="pl-pds">"</span></span>), <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
</pre></div>

<h2><a id="user-content-missing-sgrnas-in-datasets" class="anchor" href="#missing-sgrnas-in-datasets" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Missing sgRNAs in Datasets</h2>

<p>Information of how many sgRNA per gene were not present in the mapped datasets is stored in:  </p>

<p><code>r paste("_", datapath, "/", "_", sep="")</code>  </p>

<p><code>r paste("__", paste(analysis.name, "DROPOUT.xls", sep="_") ,"__", sep="")</code>  </p>

<p>In brief, the following number of sgRNAs had a <em>read count of 0</em> in the dataset: </p>

<p><strong><code>r d.CONTROL1</code>:</strong> <strong><code>r sum(U1.unmapped[,2])</code></strong> missing sgRNAs  </p>

<p><strong><code>r d.CONTROL2</code>:</strong> <strong><code>r sum(U2.unmapped[,2])</code></strong> missing sgRNAs  </p>

<p><strong><code>r d.TREAT1</code>:</strong> <strong><code>r sum(T1.unmapped[,2])</code></strong> missing sgRNAs  </p>

<p><strong><code>r d.TREAT2</code>:</strong> <strong><code>r sum(T2.unmapped[,2])</code></strong> missing sgRNAs  </p>

<h2><a id="user-content-all-stats" class="anchor" href="#all-stats" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>All Stats</h2>

<p>In depth dataset read count stats can be found in  </p>

<p><code>r paste("_", datapath, "/", "_" , sep="")</code>  </p>

<p><code>r paste("__", paste(analysis.name, "STATS.xls", sep="-") ,"__", sep="" )</code></p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">U1.allstats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">CONTROL1</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

<span class="pl-v">U2.allstats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">CONTROL2</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

<span class="pl-v">T1.allstats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">TREAT1</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)

<span class="pl-v">T2.allstats</span> <span class="pl-k">=</span> stats.data(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">TREAT2</span>, <span class="pl-v">namecolumn</span> <span class="pl-k">=</span> <span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span> <span class="pl-k">=</span> <span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dataset<span class="pl-pds">"</span></span>)


<span class="pl-v">combined.stats.mean</span> <span class="pl-k">=</span> cbind.data.frame(<span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>Name<span class="pl-pds">"</span></span>], <span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.mean<span class="pl-pds">"</span></span>],<span class="pl-smi">U2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.mean<span class="pl-pds">"</span></span>],<span class="pl-smi">T1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.mean<span class="pl-pds">"</span></span>],<span class="pl-smi">T2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.mean<span class="pl-pds">"</span></span>])
colnames(<span class="pl-smi">combined.stats.mean</span>) <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span>,<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>)

<span class="pl-v">combined.stats.median</span> <span class="pl-k">=</span> cbind.data.frame(<span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>Name<span class="pl-pds">"</span></span>], <span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.median<span class="pl-pds">"</span></span>],<span class="pl-smi">U2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.median<span class="pl-pds">"</span></span>],<span class="pl-smi">T1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.median<span class="pl-pds">"</span></span>],<span class="pl-smi">T2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.median<span class="pl-pds">"</span></span>])
colnames(<span class="pl-smi">combined.stats.median</span>) <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span>,<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>)

<span class="pl-v">combined.stats.min</span> <span class="pl-k">=</span> cbind.data.frame(<span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>Name<span class="pl-pds">"</span></span>], <span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.min<span class="pl-pds">"</span></span>],<span class="pl-smi">U2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.min<span class="pl-pds">"</span></span>],<span class="pl-smi">T1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.min<span class="pl-pds">"</span></span>],<span class="pl-smi">T2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.min<span class="pl-pds">"</span></span>])
colnames(<span class="pl-smi">combined.stats.min</span>) <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span>,<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>)

<span class="pl-v">combined.stats.max</span> <span class="pl-k">=</span> cbind.data.frame(<span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>Name<span class="pl-pds">"</span></span>], <span class="pl-smi">U1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.max<span class="pl-pds">"</span></span>],<span class="pl-smi">U2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.max<span class="pl-pds">"</span></span>],<span class="pl-smi">T1.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.max<span class="pl-pds">"</span></span>],<span class="pl-smi">T2.allstats</span>[,<span class="pl-s"><span class="pl-pds">"</span>readcount.max<span class="pl-pds">"</span></span>])
colnames(<span class="pl-smi">combined.stats.max</span>) <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Gene<span class="pl-pds">"</span></span>,<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>)

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">combined.stats.mean</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>STATS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Mean Read Count<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span> ,<span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">combined.stats.median</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>STATS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Median Read Count<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">combined.stats.min</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>STATS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Minimum Read Count<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">combined.stats.max</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>STATS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Maximum Read Count<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
</pre></div>

<p>\newpage</p>

<h1><a id="user-content-quality-control" class="anchor" href="#quality-control" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Quality Control</h1>

<h2><a id="user-content-read-distribution" class="anchor" href="#read-distribution" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Read Distribution</h2>

<p>These plots show how the read count of the sgRNAs for each dataset is distributed.
Depending on the treatment stringency, e.g. in resistance or dropout screens, the data can show asymmetry. 
However, the major population should be more or less normally distributed.</p>

<div class="highlight highlight-source-r"><pre>par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))

carpools.read.distribution(<span class="pl-smi">CONTROL1</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.CONTROL1</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>) 

carpools.read.distribution(<span class="pl-smi">CONTROL1</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.CONTROL1</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>whisker<span class="pl-pds">"</span></span>) 

carpools.read.distribution(<span class="pl-smi">CONTROL2</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.CONTROL2</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>) 

carpools.read.distribution(<span class="pl-smi">CONTROL2</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.CONTROL2</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>whisker<span class="pl-pds">"</span></span>) 
<span class="pl-c"># add new page</span>
cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

carpools.read.distribution(<span class="pl-smi">TREAT1</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.TREAT1</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>) 

carpools.read.distribution(<span class="pl-smi">TREAT1</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.TREAT1</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>whisker<span class="pl-pds">"</span></span>) 

carpools.read.distribution(<span class="pl-smi">TREAT2</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.TREAT2</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>) 

carpools.read.distribution(<span class="pl-smi">TREAT2</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>,<span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">200</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">d.TREAT2</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>log2 Readcount<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># sgRNAs<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>whisker<span class="pl-pds">"</span></span>) 
</pre></div>

<p>\pagebreak</p>

<h2><a id="user-content-read-depth" class="anchor" href="#read-depth" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Read Depth</h2>

<p>The following plot shows the read count for each gene normalized to the number of sgRNAs. Spikes indicate a higher read count per sgRNA for this particular gene.<br>
One would expect no outstanding spikes within the untreated data samples, however spikes within the treated datasets indicate a read count enrichment for this particular gene.<br>
If a non-targeting control has been set in the MIACCS file, this control is highlighted in <strong>orange</strong> color.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#, fig.width=10, fig.height=20</span>
<span class="pl-c">#par(mfrow=c(4,1))</span>
carpools.read.depth(<span class="pl-v">datasets</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>,<span class="pl-smi">CONTROL2</span>,<span class="pl-smi">TREAT1</span>,<span class="pl-smi">TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span> ,<span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">dataset.names</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">d.CONTROL1</span>,<span class="pl-smi">d.CONTROL2</span>,<span class="pl-smi">d.TREAT1</span>,<span class="pl-smi">d.TREAT2</span>), <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Genes<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Read Count per sgRNA<span class="pl-pds">"</span></span>,<span class="pl-v">statistics</span><span class="pl-k">=</span><span class="pl-smi">plot.statistic</span>, <span class="pl-v">labelgenes</span> <span class="pl-k">=</span> <span class="pl-c1">NULL</span>, <span class="pl-v">controls.target</span> <span class="pl-k">=</span> <span class="pl-smi">controls.target</span>, <span class="pl-v">controls.nontarget</span><span class="pl-k">=</span><span class="pl-smi">controls.nontarget</span>)

par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))</pre></div>

<p>\pagebreak</p>

<h2><a id="user-content-designs-per-gene" class="anchor" href="#designs-per-gene" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Designs per Gene</h2>

<p>These plots provide an overview of the representation of sgRNAs per gene within your data.<br>
Depending on the number of sgRNAs per gene in the library, one would expect a representation of more than 80 % of sgRNAs per gene in the untreated samples.
Moreover, genes with a low percentage of present sgRNAs will also show a reduced readcount.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#, fig.width=10, fig.height=10</span>
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">2</span>))

<span class="pl-v">control1.readspergene</span> <span class="pl-k">=</span> carpools.reads.genedesigns(<span class="pl-smi">CONTROL1</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">"</span>% sgRNAs:<span class="pl-pds">"</span></span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>% of sgRNAs present<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># of Genes<span class="pl-pds">"</span></span>)

<span class="pl-v">control2.readspergene</span> <span class="pl-k">=</span> carpools.reads.genedesigns(<span class="pl-smi">CONTROL2</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">"</span>% sgRNAs:<span class="pl-pds">"</span></span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>% of sgRNAs present<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># of Genes<span class="pl-pds">"</span></span>)

<span class="pl-v">treat1.readspergene</span> <span class="pl-k">=</span> carpools.reads.genedesigns(<span class="pl-smi">TREAT1</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">"</span>% sgRNAs:<span class="pl-pds">"</span></span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>% of sgRNAs present<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># of Genes<span class="pl-pds">"</span></span>)

<span class="pl-v">treat2.readspergene</span> <span class="pl-k">=</span> carpools.reads.genedesigns(<span class="pl-smi">TREAT2</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-s"><span class="pl-pds">"</span>% sgRNAs:<span class="pl-pds">"</span></span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>% of sgRNAs present<span class="pl-pds">"</span></span>, <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span># of Genes<span class="pl-pds">"</span></span>)

par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))</pre></div>

<p>\newpage</p>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(<span class="pl-k">!</span>is.null(<span class="pl-smi">controls.nontarget</span>))
  { 
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c">#cat("\\blandscape","\n")</span>
  }</pre></div>

<h2><a id="user-content-controls" class="anchor" href="#controls" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Controls</h2>

<h3><a id="user-content-non-targeting" class="anchor" href="#non-targeting" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Non-Targeting</h3>

<p>Non-targeting controls are sgRNAs that do either not target the genome at all (so called random or scramble sgRNAs) or target a gene that does not show a phenotype in the screen.<br>
Therefore, the scatter for these, which are highlighted in <strong>blue</strong>, will be distributed within the main cloud of scatter points.</p>

<p><code>r if(is.null(controls.nontarget)) {print("Unfortunately, no Non-Targeting controls were set.")}</code></p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Plot non-targeting controls</span>
<span class="pl-c">#, fig.height=20, fig.width=30,</span>
<span class="pl-k">if</span>(<span class="pl-k">!</span>is.null(<span class="pl-smi">controls.nontarget</span>))
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span>Non-targeting control: <span class="pl-pds">"</span></span>, <span class="pl-smi">controls.nontarget</span> ,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)  
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    carpools.read.count.vs(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>, <span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>), <span class="pl-v">pairs</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">controls.nontarget</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
  } <span class="pl-k">else</span> {
    cat(<span class="pl-s"><span class="pl-pds">"</span>No non-targeting control has been set in the MIACCS file.<span class="pl-pds">"</span></span>)
  }
</pre></div>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(<span class="pl-k">!</span>is.null(<span class="pl-smi">controls.nontarget</span>))
  { 
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c">#cat("\\elandscape","\n")</span>
  }</pre></div>

<p>\newpage</p>

<h3><a id="user-content-positive-controls" class="anchor" href="#positive-controls" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Positive Controls</h3>

<p>SgRNAs targeting genes that will show a phenotype in the screening setup can be used as positive controls.<br>
These will show either an enrichment (in resistance screens) or a depletion (in dropout screens) in the treatment.<br>
Within the scatter</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Plot non-targeting controls</span>
<span class="pl-c">#, fig.height=20, fig.width=30</span>
<span class="pl-k">if</span>(<span class="pl-k">!</span>is.null(<span class="pl-smi">controls.target</span>))
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span>Positive Control: <span class="pl-pds">"</span></span>, as.character(<span class="pl-smi">controls.target</span>) ,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)  
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    carpools.read.count.vs(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">TREAT1.g</span>, <span class="pl-smi">TREAT2.g</span>, <span class="pl-smi">CONTROL1.g</span>, <span class="pl-smi">CONTROL2.g</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>), <span class="pl-v">pairs</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">controls.target</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
  } <span class="pl-k">else</span> {
    cat(<span class="pl-s"><span class="pl-pds">"</span>No positive control has been set in the MIACCS file.<span class="pl-pds">"</span></span>)
  }
</pre></div>

<p>\newpage</p>

<h1><a id="user-content-hit-analysis" class="anchor" href="#hit-analysis" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Hit Analysis</h1>

<p>Hit analsysis is performed using three different methods:  </p>

<ul>
<li>Wilcox</li>
<li>DESeq2</li>
<li>MAGeCK<br></li>
</ul>

<p>For each analysis method, separate plots will be created and analysis files will be written to <strong><code>r paste(analysis.name, "HIT-CALLING.xls", sep="_")</code></strong>. See below for further information.</p>

<p>The following adjusted p-values are used to determine significance levels:  </p>

<table><thead>
<tr>
<th>Method</th>
<th>p-value</th>
</tr>
</thead><tbody>
<tr>
<td>Wilcox</td>
<td><code>r sig.pval.wilcox</code></td>
</tr>
<tr>
<td>DESeq2</td>
<td><code>r sig.pval.deseq</code></td>
</tr>
<tr>
<td>MAGeCK</td>
<td><code>r sig.pval.mageck</code></td>
</tr>
</tbody></table>

<p><strong>Wilcox</strong></p>

<p>Within this approach, the read counts of all sgRNAs in one dataset are first normalized by the function set in the MIACCS file. By default, normalization is done by read count division with the dataset median.<br>
Then, the fold change of each population of sgRNAs for a gene is tested against the population of either the non-targeting controls or randomly picked sgRNAs, as defined by the random picks option within the MIACCS file, using a two-sided Mann-Whitney test with FDR correction.  </p>

<p><strong>DESeq2</strong></p>

<p>For the DESeq2 analysis implementation, the read counts of all sgRNAs for a given gene are first summed up to increase the available read count.<br>
Then, DESeq2 analysis is perfomed, which includes the estimation of size-factors, the variance stabilization using a parametric fit and a Wald-Test for differnece in log2 fold changes between the untreated and treated data.<br>
More information about this can be found in <em>Love et al.</em><br>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/25516281">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</a><br>
<em>Genome Biology</em> 2014  </p>

<p><strong>MAGeCK</strong></p>

<p>MAGeCK analysis uses a rank-based model to test for a change in abundance of sgRNAs after median normalization of the dataset.<br>
Further information can be found at the <a href="http://sourceforge.net/projects/mageck/">MAGeCK Homepage</a>.</p>

<p>\newpage</p>

<h2><a id="user-content-wilcox" class="anchor" href="#wilcox" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Wilcox</h2>

<p>All analysis data can be found in the <strong><code>r paste(analysis.name, "HIT-CALLING.xls", sep="_")</code></strong> file.  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># wilcox</span>

<span class="pl-k">if</span>(is.null(<span class="pl-smi">controls.nontarget</span>))
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(paste(<span class="pl-s"><span class="pl-pds">"</span>**Since no non-targeting controls were set,<span class="pl-pds">"</span></span>, <span class="pl-smi">control.picks</span>, <span class="pl-s"><span class="pl-pds">"</span>sgRNAs are picked from the datset as reference population.**<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>))
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
  }
<span class="pl-v">data.wilcox</span> <span class="pl-k">=</span> stat.wilcox(<span class="pl-v">untreated.list</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>), <span class="pl-v">treated.list</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">TREAT1</span>,<span class="pl-smi">TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.fun</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">sorting</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">controls</span><span class="pl-k">=</span><span class="pl-smi">controls.nontarget</span>, <span class="pl-v">control.picks</span><span class="pl-k">=</span><span class="pl-smi">control.picks</span>)

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">data.wilcox</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>HIT-CALLING.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Wilcox<span class="pl-pds">"</span></span>)
</pre></div>

<h3><a id="user-content-p-value-distribution" class="anchor" href="#p-value-distribution" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>P-Value Distribution</h3>

<p>For all genes present in the data, the -log10 corrected p-values are plotted to estimate how well the analysis method performed for this screen. A straight line of points with only small differences in the p-value indicates that the analysis method did not perform well. 
Genes that resulted in a p-value below the threshold set in the MIACCS file are highlighted in red color.  </p>

<div class="highlight highlight-source-r"><pre>carpools.waterfall.pval(<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>wilcox<span class="pl-pds">"</span></span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">pval</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">log</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre></div>

<h3><a id="user-content-enriched" class="anchor" href="#enriched" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enriched</h3>

<p>The following genes showed enrichment in the treatment datasets with a <strong>p-value smaller than <code>r sig.pval.wilcox</code></strong>:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.wilcox.plot.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox</span>[<span class="pl-smi">data.wilcox</span><span class="pl-k">$</span><span class="pl-smi">foldchange</span> <span class="pl-k">&gt;</span> <span class="pl-c1">1</span>,]
<span class="pl-v">data.wilcox.plot.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox.plot.enriched</span>[order(<span class="pl-smi">data.wilcox.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">p.value</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-k">if</span>(nrow(<span class="pl-smi">data.wilcox.plot.enriched</span>[<span class="pl-smi">data.wilcox.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">p.value</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.wilcox</span>,]) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
 <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.wilcox.plot.enriched</span>[<span class="pl-smi">data.wilcox.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">p.value</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.wilcox</span>,])
  } <span class="pl-k">else</span> { cat(<span class="pl-s"><span class="pl-pds">"</span>**No genes showed significant enrichment with a p-value lower than<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>)}

</pre></div>

<p>According to the value set, the <strong>top <code>r number.hits.plot.enriched</code> % enriched genes</strong> were:  </p>

<p><code>r knitr::kable(data.wilcox.plot.enriched[1:number.enriched.counter,])</code></p>

<p>\newpage</p>

<h3><a id="user-content-depleted" class="anchor" href="#depleted" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Depleted</h3>

<p>The following genes showed depletion in the treatment datasets with a <strong>p-value smaller than <code>r sig.pval.wilcox</code></strong>:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.wilcox.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox</span>[<span class="pl-smi">data.wilcox</span><span class="pl-k">$</span><span class="pl-smi">foldchange</span> <span class="pl-k">&lt;</span> <span class="pl-c1">1</span>,]
<span class="pl-v">data.wilcox.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox.plot.depleted</span>[order(<span class="pl-smi">data.wilcox.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">p.value</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-k">if</span>(nrow(<span class="pl-smi">data.wilcox.plot.depleted</span>[<span class="pl-smi">data.wilcox.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">p.value</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.wilcox</span>,]) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
 <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.wilcox.plot.depleted</span>[<span class="pl-smi">data.wilcox.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">p.value</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.wilcox</span>,])
  } <span class="pl-k">else</span> { cat(<span class="pl-s"><span class="pl-pds">"</span>**No genes showed significant depletion with a p-value lower than<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>)}
</pre></div>

<p>According to the value set, the <strong>top <code>r number.hits.plot.depleted</code> % depleted genes</strong> were:  </p>

<p><code>r knitr::kable(data.wilcox.plot.depleted[1:number.depleted.counter,])</code></p>

<p>\newpage</p>

<h2><a id="user-content-deseq2" class="anchor" href="#deseq2" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>DESeq2</h2>

<p>All analysis data can be found in the <strong><code>r paste(analysis.name, "HIT-CALLING.xls", sep="_")</code></strong> file.  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># DESeq2</span>
<span class="pl-v">data.deseq</span> <span class="pl-k">=</span> stat.DESeq(<span class="pl-v">untreated.list</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>), <span class="pl-v">treated.list</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">TREAT1</span>,<span class="pl-smi">TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">sorting</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">filename.deseq</span> <span class="pl-k">=</span> paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>-ANALYSIS-DESeq2-sgRNA.tab<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">sgRNA.pval</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">fitType</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>mean<span class="pl-pds">"</span></span>)

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(as.data.frame(<span class="pl-smi">data.deseq</span><span class="pl-k">$</span><span class="pl-smi">genes</span>), <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>HIT-CALLING.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>DESeq2<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
</pre></div>

<h3><a id="user-content-p-value-distribution-1" class="anchor" href="#p-value-distribution-1" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>P-Value Distribution</h3>

<p>For all genes present in the data, the -log10 corrected p-values are plotted to estimate how well the analysis method performed for this screen. A straight line of points with only small differences in the p-value indicates that the analysis method did not perform well. 
Genes that resulted in a p-value below the threshold set in the MIACCS file are highlighted in red color.  </p>

<div class="highlight highlight-source-r"><pre>carpools.waterfall.pval(<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>deseq2<span class="pl-pds">"</span></span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">pval</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">log</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre></div>

<h3><a id="user-content-enriched-1" class="anchor" href="#enriched-1" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enriched</h3>

<p>The following genes showed enrichment in the treatment datasets with a <strong>p-value smaller than <code>r sig.pval.deseq</code></strong>:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.deseq.plot.enriched</span> <span class="pl-k">=</span> as.data.frame(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">log2FoldChange</span> <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>,])
<span class="pl-v">data.deseq.plot.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq.plot.enriched</span>[order(<span class="pl-smi">data.deseq.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">padj</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-k">if</span>(nrow(<span class="pl-smi">data.deseq.plot.enriched</span>[<span class="pl-smi">data.deseq.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">padj</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.deseq</span>,c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">6</span>,<span class="pl-c1">8</span>)]) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
 <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.deseq.plot.enriched</span>[<span class="pl-smi">data.deseq.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">padj</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.deseq</span>,c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">6</span>,<span class="pl-c1">8</span>)])
  } <span class="pl-k">else</span> { cat(<span class="pl-s"><span class="pl-pds">"</span>**No genes showed significant enrichment with a p-value lower than<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>)}</pre></div>

<p>According to the value set, the <strong>top <code>r number.hits.plot.enriched</code> % enriched genes</strong> were:  </p>

<p><code>r knitr::kable(data.deseq.plot.enriched[1:number.enriched.counter,c(2,3,6,8)])</code></p>

<p>\newpage</p>

<h3><a id="user-content-depleted-1" class="anchor" href="#depleted-1" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Depleted</h3>

<p>The following genes showed depletion in the treatment datasets with a <strong>p-value smaller than <code>r sig.pval.deseq</code></strong>:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.deseq.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">log2FoldChange</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0</span>,]
<span class="pl-v">data.deseq.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq.plot.depleted</span>[order(<span class="pl-smi">data.deseq.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">padj</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-k">if</span>(nrow(<span class="pl-smi">data.deseq.plot.depleted</span>[<span class="pl-smi">data.deseq.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">padj</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.deseq</span>,c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">6</span>,<span class="pl-c1">8</span>)]) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
 <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.deseq.plot.depleted</span>[<span class="pl-smi">data.deseq.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">padj</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.deseq</span>,c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">6</span>,<span class="pl-c1">8</span>)])
  } <span class="pl-k">else</span> { cat(<span class="pl-s"><span class="pl-pds">"</span>**No genes showed significant depletion with a p-value lower than<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>)}

</pre></div>

<p>According to the value set, the <strong>top <code>r number.hits.plot.depleted</code> % depleted genes</strong> were:  </p>

<p><code>r knitr::kable(data.deseq.plot.depleted[1:number.depleted.counter,c(2,3,6,8)])</code></p>

<p>\newpage</p>

<h2><a id="user-content-mageck" class="anchor" href="#mageck" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>MAGeCK</h2>

<p>All analysis data for MAGeCK can be found in the <strong><code>r paste(analysis.name, "HIT-CALLING.xls", sep="_")</code></strong> file.  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># MAGeCK</span>
<span class="pl-v">data.mageck</span> <span class="pl-k">=</span> stat.mageck(<span class="pl-v">untreated.list</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>), <span class="pl-v">treated.list</span> <span class="pl-k">=</span> <span class="pl-k">list</span>(<span class="pl-smi">TREAT1</span>,<span class="pl-smi">TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">norm.fun</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>median<span class="pl-pds">"</span></span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">mageckfolder</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">sort.criteria</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>neg<span class="pl-pds">"</span></span>, <span class="pl-v">adjust.method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>fdr<span class="pl-pds">"</span></span>, <span class="pl-v">filename</span> <span class="pl-k">=</span> paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>-ANALYSIS-MAGeCK-RAW<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>), <span class="pl-v">fdr.pval</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.mageck</span>)

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(as.data.frame(<span class="pl-smi">data.mageck</span><span class="pl-k">$</span><span class="pl-smi">genes</span>), <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>HIT-CALLING.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>MAGeCK<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-c"># get number of sig gene</span>
<span class="pl-v">mageck.cutoff</span> <span class="pl-k">=</span> nrow(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">pos</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.mageck</span>,]) <span class="pl-k">+</span> nrow(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">neg</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.mageck</span>,])</pre></div>

<h3><a id="user-content-p-value-distribution-2" class="anchor" href="#p-value-distribution-2" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>P-Value Distribution</h3>

<p>For all genes present in the data, the -log10 corrected p-values are plotted to estimate how well the analysis method performed for this screen. A straight line of points with only small differences in the p-value indicates that the analysis method did not perform well. 
Genes that resulted in a p-value below the threshold set in the MIACCS file are highlighted in red color.  </p>

<div class="highlight highlight-source-r"><pre>carpools.waterfall.pval(<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">pval</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">log</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre></div>

<h3><a id="user-content-enriched-2" class="anchor" href="#enriched-2" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enriched</h3>

<p>The following genes showed enrichment in the treatment datasets with <strong>a p-value smaller than <code>r sig.pval.mageck</code></strong>:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.mageck.plot.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">pos</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.mageck</span>,]
<span class="pl-v">data.mageck.plot.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck.plot.enriched</span>[order(<span class="pl-smi">data.mageck.plot.enriched</span><span class="pl-k">$</span><span class="pl-smi">rank.pos</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-k">if</span>(nrow(<span class="pl-smi">data.mageck.plot.enriched</span>) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
 <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.mageck.plot.enriched</span>[,c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">4</span>,<span class="pl-c1">5</span>,<span class="pl-c1">6</span>,<span class="pl-c1">7</span>)])
  } <span class="pl-k">else</span> { cat(<span class="pl-s"><span class="pl-pds">"</span>**No genes showed significant enrichment with a p-value lower than<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>)}

<span class="pl-v">data.mageck.plot.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][order(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">rank.pos</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]</pre></div>

<p>According to the value set, the <strong>top <code>r number.hits.plot.enriched</code> % enriched genes</strong> were:  </p>

<p><code>r knitr::kable(data.mageck.plot.enriched[1:number.enriched.counter,c(2,3,4,5,6,7)])</code></p>

<p>\newpage</p>

<h3><a id="user-content-depleted-2" class="anchor" href="#depleted-2" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Depleted</h3>

<p>The following genes showed depletion in the treatment datasets with a <strong>p-value smaller than <code>r sig.pval.mageck</code></strong>:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.mageck.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">neg</span> <span class="pl-k">&lt;</span> <span class="pl-smi">sig.pval.mageck</span>,]
<span class="pl-v">data.mageck.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck.plot.depleted</span>[order(<span class="pl-smi">data.mageck.plot.depleted</span><span class="pl-k">$</span><span class="pl-smi">rank.neg</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]
<span class="pl-k">if</span>(nrow(<span class="pl-smi">data.mageck.plot.depleted</span>) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
  <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.mageck.plot.depleted</span>[,c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">4</span>,<span class="pl-c1">5</span>,<span class="pl-c1">6</span>,<span class="pl-c1">7</span>)])
  } <span class="pl-k">else</span> { cat(<span class="pl-s"><span class="pl-pds">"</span>**No genes showed significant depletion with a p-value lower than<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>)}

<span class="pl-v">data.mageck.plot.depleted</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][order(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]<span class="pl-k">$</span><span class="pl-smi">rank.neg</span>, <span class="pl-v">na.last</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>),]</pre></div>

<p>According to the value set, the <strong>top <code>r number.hits.plot.depleted</code> % depleted genes</strong> were:  </p>

<p><code>r knitr::kable(data.mageck.plot.depleted[1:number.depleted.counter,c(2,3,4,5,6,7)])</code></p>

<p>\newpage</p>

<h1><a id="user-content-hit-candidate-overview" class="anchor" href="#hit-candidate-overview" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Hit Candidate Overview</h1>

<h2><a id="user-content-overview" class="anchor" href="#overview" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Overview</h2>

<p>Genes which showed enrichment or depletion within the individual analysis methods are presented in the following section.<br>
All genes that showed <em>significant</em> <strong>enrichment</strong> within Wilcox, DESeq2 and MAGeCK with the given p-value cutoffs are highlighted in <strong>red</strong> color.<br>
All genes that showed <em>significant</em> <strong>depletion</strong> within Wilcox, DESeq2 and MAGeCK with the given p-value cutoffs are highlighted in <strong>blue</strong> color.  </p>

<p>Moreover, all genes that showed up as <em>significantly</em> enriched or depleted in any of the analysis methods are highlighted in <strong>orange</strong> color.
Genes that showed no significant effect are presented in grey or black color.  </p>

<div class="highlight highlight-source-r"><pre>carpools.hit.overview(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">plot.genes</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>overlapping<span class="pl-pds">"</span></span>)
</pre></div>

<div class="highlight highlight-source-r"><pre><span class="pl-v">overlap.enriched</span> <span class="pl-k">=</span> generate.hits(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">compare.cutoff</span>, <span class="pl-v">plot.genes</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>overlapping<span class="pl-pds">"</span></span>)

<span class="pl-v">overlap.depleted</span> <span class="pl-k">=</span> generate.hits(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">compare.cutoff</span>, <span class="pl-v">plot.genes</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>overlapping<span class="pl-pds">"</span></span>)

<span class="pl-c"># Combine datasets for overlapping hits</span>
<span class="pl-v">overlap.enriched.dataset</span> <span class="pl-k">=</span> <span class="pl-k">data.frame</span>(
  <span class="pl-v">gene</span> <span class="pl-k">=</span> <span class="pl-smi">overlap.enriched</span>,
  <span class="pl-v">log2FC</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>, <span class="pl-s"><span class="pl-pds">"</span>log2FoldChange<span class="pl-pds">"</span></span>],
  <span class="pl-v">wilcox.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox</span>[rownames(<span class="pl-smi">data.wilcox</span>) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>,<span class="pl-s"><span class="pl-pds">"</span>p.value<span class="pl-pds">"</span></span>],
  <span class="pl-v">deseq2.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>,<span class="pl-s"><span class="pl-pds">"</span>padj<span class="pl-pds">"</span></span>],
  <span class="pl-v">mageck.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>,<span class="pl-s"><span class="pl-pds">"</span>pos<span class="pl-pds">"</span></span>],
  <span class="pl-v">stringsAsFactors</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

<span class="pl-v">overlap.depleted.dataset</span> <span class="pl-k">=</span> <span class="pl-k">data.frame</span>(
  <span class="pl-v">gene</span> <span class="pl-k">=</span> <span class="pl-smi">overlap.depleted</span>,
  <span class="pl-v">log2FC</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>, <span class="pl-s"><span class="pl-pds">"</span>log2FoldChange<span class="pl-pds">"</span></span>],
  <span class="pl-v">wilcox.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox</span>[rownames(<span class="pl-smi">data.wilcox</span>) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>,<span class="pl-s"><span class="pl-pds">"</span>p.value<span class="pl-pds">"</span></span>],
  <span class="pl-v">deseq2.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>,<span class="pl-s"><span class="pl-pds">"</span>padj<span class="pl-pds">"</span></span>],
  <span class="pl-v">mageck.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>,<span class="pl-s"><span class="pl-pds">"</span>neg<span class="pl-pds">"</span></span>],
  <span class="pl-v">stringsAsFactors</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)</pre></div>

<p>\newpage</p>

<h2><a id="user-content-overlaps-in-enrichment-analysis" class="anchor" href="#overlaps-in-enrichment-analysis" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Overlaps in Enrichment Analysis</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(nrow(<span class="pl-smi">overlap.enriched.dataset</span>) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span>The following genes showed enrichment in all three analysis methods.<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">overlap.enriched.dataset</span>)
    <span class="pl-v">cutoff.override.enriched</span> <span class="pl-k">=</span> <span class="pl-smi">cutoff.override</span>

  } <span class="pl-k">else</span> {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>**ATTENTION**  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>**No overlapping enriched genes were found between all three methods.** <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>Therefore, the overlap from the<span class="pl-pds">"</span></span>, <span class="pl-smi">number.hits.plot.enriched</span>, <span class="pl-s"><span class="pl-pds">"</span>% of genes within all methods is used for plotting possible gene candidates.<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span> )
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>We strongly advise you to have a closer look at the individual outputs of the analysis methods and to carefully look at the following candidate genes.<span class="pl-pds">"</span></span>)

    <span class="pl-v">cutoff.override.enriched</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>

    <span class="pl-v">overlap.enriched</span> <span class="pl-k">=</span> generate.hits(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span> <span class="pl-smi">cutoff.override.enriched</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.enriched.counter</span>, <span class="pl-v">plot.genes</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>overlapping<span class="pl-pds">"</span></span>)

    <span class="pl-v">overlap.enriched.dataset</span> <span class="pl-k">=</span> <span class="pl-k">data.frame</span>(
  <span class="pl-v">gene</span> <span class="pl-k">=</span> <span class="pl-smi">overlap.enriched</span>,
  <span class="pl-v">log2FC</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>, <span class="pl-s"><span class="pl-pds">"</span>log2FoldChange<span class="pl-pds">"</span></span>],
  <span class="pl-v">wilcox.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox</span>[rownames(<span class="pl-smi">data.wilcox</span>) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>,<span class="pl-s"><span class="pl-pds">"</span>p.value<span class="pl-pds">"</span></span>],
  <span class="pl-v">deseq2.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>,<span class="pl-s"><span class="pl-pds">"</span>padj<span class="pl-pds">"</span></span>],
  <span class="pl-v">mageck.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.enriched</span>,<span class="pl-s"><span class="pl-pds">"</span>pos<span class="pl-pds">"</span></span>],
  <span class="pl-v">stringsAsFactors</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

  }
</pre></div>

<p>Within the <strong>top enriched hits</strong>, the overlap of enriched hits per analysis method is displayed as follows:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">venn.enriched</span> <span class="pl-k">=</span> compare.analysis(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.enriched</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.enriched.counter</span>, <span class="pl-v">output</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>venn<span class="pl-pds">"</span></span>)

require(<span class="pl-smi">VennDiagram</span>)
<span class="pl-e">grid</span><span class="pl-k">::</span>grid.draw(<span class="pl-e">VennDiagram</span><span class="pl-k">::</span>venn.diagram(<span class="pl-smi">venn.enriched</span>, <span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">fill</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>lightgreen<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>lightblue2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>lightgray<span class="pl-pds">"</span></span>), <span class="pl-v">na</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>remove<span class="pl-pds">"</span></span>, <span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>, <span class="pl-v">cat.cex</span><span class="pl-k">=</span><span class="pl-c1">2</span>))</pre></div>

<p>\newpage</p>

<h2><a id="user-content-overlaps-in-depletion-analysis" class="anchor" href="#overlaps-in-depletion-analysis" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Overlaps in Depletion Analysis</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(nrow(<span class="pl-smi">overlap.depleted.dataset</span>) <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>)
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span>The following genes showed depletion in all three analysis methods.<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">overlap.depleted.dataset</span>)
    <span class="pl-v">cutoff.override.depleted</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override</span>

  } <span class="pl-k">else</span> {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>**ATTENTION**  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>**No overlapping depleted genes were found between all three methods.**<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>Therefore, the overlap from the<span class="pl-pds">"</span></span>, <span class="pl-smi">number.hits.plot.depleted</span>, <span class="pl-s"><span class="pl-pds">"</span>% of genes within all methods is used for plotting possible hit candidates.<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span> )
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>We strongly advise you to have a closer look at the individual outputs of the analysis methods and to carefully look at the following candidate genes.<span class="pl-pds">"</span></span>)

    <span class="pl-v">cutoff.override.depleted</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>

    <span class="pl-v">overlap.depleted</span> <span class="pl-k">=</span> generate.hits(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.depleted</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.depleted.counter</span>, <span class="pl-v">plot.genes</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>overlapping<span class="pl-pds">"</span></span>)

    <span class="pl-v">overlap.depleted.dataset</span> <span class="pl-k">=</span> <span class="pl-k">data.frame</span>(
  <span class="pl-v">gene</span> <span class="pl-k">=</span> <span class="pl-smi">overlap.depleted</span>,
  <span class="pl-v">log2FC</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>, <span class="pl-s"><span class="pl-pds">"</span>log2FoldChange<span class="pl-pds">"</span></span>],
  <span class="pl-v">wilcox.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.wilcox</span>[rownames(<span class="pl-smi">data.wilcox</span>) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>,<span class="pl-s"><span class="pl-pds">"</span>p.value<span class="pl-pds">"</span></span>],
  <span class="pl-v">deseq2.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.deseq</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>,<span class="pl-s"><span class="pl-pds">"</span>padj<span class="pl-pds">"</span></span>],
  <span class="pl-v">mageck.pval</span> <span class="pl-k">=</span> <span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]][rownames(<span class="pl-smi">data.mageck</span>[[<span class="pl-c1">1</span>]]) <span class="pl-k">%in%</span> <span class="pl-smi">overlap.depleted</span>,<span class="pl-s"><span class="pl-pds">"</span>neg<span class="pl-pds">"</span></span>],
  <span class="pl-v">stringsAsFactors</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

  }
</pre></div>

<p>Within the <strong>top depleted hits</strong>, the overlap of depleted hits per analysis method is displayed as follows:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">venn.depleted</span> <span class="pl-k">=</span> compare.analysis(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.depleted</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.depleted.counter</span>, <span class="pl-v">output</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>venn<span class="pl-pds">"</span></span>)

require(<span class="pl-smi">VennDiagram</span>)
<span class="pl-e">grid</span><span class="pl-k">::</span>grid.draw(<span class="pl-e">VennDiagram</span><span class="pl-k">::</span>venn.diagram(<span class="pl-smi">venn.depleted</span>, <span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">fill</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>lightgreen<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>lightblue2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>lightgray<span class="pl-pds">"</span></span>), <span class="pl-v">na</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>remove<span class="pl-pds">"</span></span>, <span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">2</span>,<span class="pl-v">lty</span><span class="pl-k">=</span><span class="pl-c1">2</span>, <span class="pl-v">cat.cex</span><span class="pl-k">=</span><span class="pl-c1">2</span>))</pre></div>

<p>\newpage</p>

<h2><a id="user-content-enriched-overlapping-hit-candidates" class="anchor" href="#enriched-overlapping-hit-candidates" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enriched Overlapping Hit Candidates</h2>

<p>All overlapping genes which showed enrichment are highlighted in <strong>orange</strong> color in the following plots.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># make Hit Scatter with all HITS enriched in a 4x4 square</span>

carpools.read.count.vs(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">CONTROL1.g</span>, <span class="pl-smi">CONTROL2.g</span>,<span class="pl-smi">TREAT1.g</span>, <span class="pl-smi">TREAT2.g</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>), <span class="pl-v">pairs</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">overlap.enriched</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre></div>

<h2><a id="user-content-depleted-overlapping-hit-candidates" class="anchor" href="#depleted-overlapping-hit-candidates" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Depleted Overlapping Hit Candidates</h2>

<p>All overlapping genes which showed depletion are highlighted in <strong>orange</strong> color in the following plots.</p>

<div class="highlight highlight-source-r"><pre>carpools.read.count.vs(<span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">CONTROL1.g</span>, <span class="pl-smi">CONTROL2.g</span>,<span class="pl-smi">TREAT1.g</span>, <span class="pl-smi">TREAT2.g</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>), <span class="pl-v">pairs</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">overlap.depleted</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)</pre></div>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(identical(<span class="pl-smi">g.convert</span>,<span class="pl-c1">TRUE</span>))
  {
    <span class="pl-v">g.identifier</span> <span class="pl-k">=</span> <span class="pl-smi">g.identifier.new</span> 
  }

<span class="pl-k">if</span>(identical(<span class="pl-smi">a.annotate.hits</span>,<span class="pl-c1">TRUE</span>))
  {
  <span class="pl-v">annotate.enriched</span> <span class="pl-k">=</span> get.gene.info(as.data.frame(<span class="pl-smi">overlap.enriched</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>ensembl_gene_id<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>hgnc_symbol<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>description<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>name_1006<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>mim_gene_description<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>family_description<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_peptide_id<span class="pl-pds">"</span></span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>info<span class="pl-pds">"</span></span>, <span class="pl-v">controls</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)


  <span class="pl-v">annotate.depleted</span> <span class="pl-k">=</span> get.gene.info(as.data.frame(<span class="pl-smi">overlap.depleted</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>ensembl_gene_id<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>hgnc_symbol<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>description<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>name_1006<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>mim_gene_description<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>family_description<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_peptide_id<span class="pl-pds">"</span></span>), <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>info<span class="pl-pds">"</span></span>, <span class="pl-v">controls</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

  }</pre></div>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># create final table with all genes + hit analysis results</span>
<span class="pl-v">final.tab</span> <span class="pl-k">=</span> final.table(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">CONTROL1.g</span>, <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>genes<span class="pl-pds">"</span></span>)

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">final.tab</span>, paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>FINAL.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Genelist<span class="pl-pds">"</span></span>, 
  <span class="pl-v">col.names</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">showNA</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

<span class="pl-c">#final.tab.sgRNA = final.table(wilcox=data.wilcox, deseq=data.deseq, mageck=data.mageck, dataset=CONTROL1.g, namecolumn=namecolumn, type="all")</span>

<span class="pl-c">#xlsx::write.xlsx(final.tab, paste(datapath, paste(analysis.name, "FINAL.xls", sep="_"), sep="/"), sheetName="All", </span>
  <span class="pl-c">#col.names=TRUE, row.names=FALSE, append=TRUE, showNA=TRUE)</span></pre></div>

<p>\newpage</p>

<h1><a id="user-content-hit-candidates" class="anchor" href="#hit-candidates" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Hit Candidates</h1>

<p>Scatterplots representing the gene read count as well as sgRNA read count for all overlapping hits are plotted in this section.</p>

<hr>

<p>If there were <em>signficantly</em> enriched or depleted genes that overlapped in all analysis methods, they will be presented here.<br>
Therefore, the top overlapping hits of each Hit Analysis with a p-value below the thresholds<br>
for each analysis method are highlighted in <code>r plot.labelcolor</code> within the scatter plots for all four samples.  </p>

<p>In the case that no <em>significantly</em> enriched or depleted genes did overlap in all methods, those that overlapped within the <code>r number.hits.plot.enriched</code> % of top enriched and the top <code>r number.hits.plot.depleted</code> % depleted genes are used.<br>
Therefore, the top overlapping hits of each Hit Analysis are highlighted in <code>r plot.labelcolor</code> within the scatter plots for all four samples.</p>

<hr>

<p>This allows a fast and easy view for single genes and its individual sgRNAs.<br>
Moreover, individual sgRNA effects are plotted as well as the corresponding target sequence.  </p>

<p>In this section, the following plots are <strong>generated for each hit candidate</strong>:  </p>

<ul>
<li>Scatterplot with gene read count within all datasets</li>
<li>Scatterplot with sgRNA read count within all datasets</li>
<li>sgRNA log2 fold changes</li>
<li>sgRNA log2 fold change distribution in comparison to all sgRNAs or given controls</li>
<li>sgRNA target sequence list</li>
</ul>

<p>The scatter plots show the median normalized, log read count of each genes/sgRNA.
Moreover, the <strong>blue lines indicate a read count foldchange of 2</strong>, the <strong>green lines indicate a read count foldchange of 4</strong>.</p>

<p>\newpage</p>

<h2><a id="user-content-enriched-3" class="anchor" href="#enriched-3" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enriched</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># get overlap.enriched as list of genes</span>
<span class="pl-v">count</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>
<span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-smi">overlap.enriched</span>)
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>###<span class="pl-pds">"</span></span>, <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    <span class="pl-c"># Prepare Output of Gene Annotation</span>
    <span class="pl-k">if</span>(identical(<span class="pl-smi">a.annotate.hits</span>,<span class="pl-c1">TRUE</span>))
      {
      cat(<span class="pl-s"><span class="pl-pds">"</span>In addition to the plots below, the following information has been retrieved via biomaRt:  <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
      cat(<span class="pl-s"><span class="pl-pds">"</span>__ENSEMBL ID__ (links to Ensembl)<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span> ,paste(<span class="pl-s"><span class="pl-pds">"</span>[<span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_gene_id<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>]<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>http://www.ensembl.org/id/<span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_gene_id<span class="pl-pds">"</span></span>], <span class="pl-s"><span class="pl-pds">"</span>)<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>),<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__HGNC SYMBOL__ (links to GeneCards)<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,paste(<span class="pl-s"><span class="pl-pds">"</span>[<span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>hgnc_symbol<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>]<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>http://www.genecards.org/cgi-bin/carddisp.pl?gene=<span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>hgnc_symbol<span class="pl-pds">"</span></span>], <span class="pl-s"><span class="pl-pds">"</span>)<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>) ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__GENE DESCRIPTION__ <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>description<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__GO TERM__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>name_1006<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__MIM GENE DESCRIPTION__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>mim_gene_description<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__ENSEMBL PROTEIN ID__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_peptide_id<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__PROTEIN FAMILY DESCRIPTION__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.enriched</span>[<span class="pl-smi">annotate.enriched</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>family_description<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c"># Significantly?</span>
    <span class="pl-v">sig.pval.candidate</span> <span class="pl-k">=</span> <span class="pl-smi">final.tab</span>[<span class="pl-smi">final.tab</span><span class="pl-k">$</span><span class="pl-smi">name</span> <span class="pl-k">==</span> <span class="pl-smi">i</span>,c(<span class="pl-s"><span class="pl-pds">"</span>wilcox.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>deseq2.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mageck.fdr.pos<span class="pl-pds">"</span></span>)]
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Wilcox p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">1</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__DESeq2 p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">2</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__MAGeCK p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">3</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)  
      }
    <span class="pl-k">else</span>
      {
        <span class="pl-c"># Significantly?</span>
    <span class="pl-v">sig.pval.candidate</span> <span class="pl-k">=</span> <span class="pl-smi">final.tab</span>[<span class="pl-smi">final.tab</span><span class="pl-k">$</span><span class="pl-smi">name</span> <span class="pl-k">==</span> <span class="pl-smi">i</span>,c(<span class="pl-s"><span class="pl-pds">"</span>wilcox.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>deseq2.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mageck.fdr.pos<span class="pl-pds">"</span></span>)]
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Wilcox p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">1</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__DESeq2 p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">2</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__MAGeCK p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">3</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
      }

    <span class="pl-c"># Scatterplot PAIRS GENES</span>
    <span class="pl-v">plothitsscatter.enriched</span> <span class="pl-k">=</span> carpools.hit.scatter(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">TREAT1.g</span>, <span class="pl-smi">TREAT2.g</span>, <span class="pl-smi">CONTROL1.g</span>, <span class="pl-smi">CONTROL2.g</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-smi">plot.labelcolor</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">median</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-smi">offsetplot</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.enriched</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>,  <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>)

    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c">#cat("\\newpage","\n")</span>
    <span class="pl-c">#cat("\n")</span>

    <span class="pl-c"># Scatterplot PAIRS sgRNA</span>
    <span class="pl-v">plothitsscatter.enriched</span> <span class="pl-k">=</span> carpools.hit.scatter(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>, <span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-smi">plot.labelcolor</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">median</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-smi">offsetplot</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.enriched</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>,  <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>)

    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    <span class="pl-v">sgrnas.en</span> <span class="pl-k">=</span> carpools.hit.sgrna(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>, <span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">put.names</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>, <span class="pl-v">controls.target</span><span class="pl-k">=</span><span class="pl-smi">controls.target</span>, <span class="pl-v">controls.nontarget</span><span class="pl-k">=</span><span class="pl-smi">controls.nontarget</span>)

    par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))

    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    <span class="pl-c"># sgRNA table</span>
    <span class="pl-k">if</span>(<span class="pl-smi">count</span><span class="pl-k">==</span><span class="pl-c1">0</span>)
      {
      <span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">overlap.enriched</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, paste(<span class="pl-s"><span class="pl-pds">"</span>HITS-sgRNA<span class="pl-pds">"</span></span>, paste(<span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>List of Genes<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">col.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
      }

  <span class="pl-v">sgrnas.en.table</span> <span class="pl-k">=</span> carpools.sgrna.table(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>, <span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.enriched</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>, <span class="pl-v">sgrna.file</span> <span class="pl-k">=</span> <span class="pl-smi">libFILE</span>, <span class="pl-v">write</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

  <span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">sgrnas.en.table</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, paste(<span class="pl-s"><span class="pl-pds">"</span>HITS-sgRNA<span class="pl-pds">"</span></span>, paste(<span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span>as.character(<span class="pl-smi">i</span>), <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)


print(<span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">sgrnas.en.table</span>))
  cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
  cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

<span class="pl-v">count</span> <span class="pl-k">=</span> <span class="pl-smi">count</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>
}
</pre></div>

<p>\newpage</p>

<div class="highlight highlight-source-r"><pre>cat(<span class="pl-s"><span class="pl-pds">"</span>##<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Depleted <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
<span class="pl-v">count</span><span class="pl-k">=</span><span class="pl-c1">0</span>
<span class="pl-c"># get overlap.enriched as list of genes</span>
<span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-smi">overlap.depleted</span>)
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>###<span class="pl-pds">"</span></span>, <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    <span class="pl-k">if</span>(identical(<span class="pl-smi">a.annotate.hits</span>,<span class="pl-c1">TRUE</span>))
      {
      <span class="pl-c"># Prepare Output of Gene Annotation</span>
    cat(<span class="pl-s"><span class="pl-pds">"</span>In addition to the plots below, the following information has been retrieved via biomaRt:  <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
      cat(<span class="pl-s"><span class="pl-pds">"</span>__ENSEMBL ID__ (links to Ensembl)<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span> ,paste(<span class="pl-s"><span class="pl-pds">"</span>[<span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_gene_id<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>]<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>http://www.ensembl.org/id/<span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_gene_id<span class="pl-pds">"</span></span>], <span class="pl-s"><span class="pl-pds">"</span>)<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>),<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__HGNC SYMBOL__ (links to GeneCards)<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,paste(<span class="pl-s"><span class="pl-pds">"</span>[<span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>hgnc_symbol<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>]<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>http://www.genecards.org/cgi-bin/carddisp.pl?gene=<span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>hgnc_symbol<span class="pl-pds">"</span></span>], <span class="pl-s"><span class="pl-pds">"</span>)<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>) ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__GENE DESCRIPTION__ <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>description<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__GO TERM__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>name_1006<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__MIM GENE DESCRIPTION__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>mim_gene_description<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__ENSEMBL PROTEIN ID__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>ensembl_peptide_id<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__PROTEIN FAMILY DESCRIPTION__ <span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-smi">annotate.depleted</span>[<span class="pl-smi">annotate.depleted</span>[,<span class="pl-c1">1</span>] <span class="pl-k">==</span> <span class="pl-smi">i</span>,<span class="pl-s"><span class="pl-pds">"</span>family_description<span class="pl-pds">"</span></span>] ,<span class="pl-s"><span class="pl-pds">"</span>  <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c"># Significantly?</span>
    <span class="pl-v">sig.pval.candidate</span> <span class="pl-k">=</span> <span class="pl-smi">final.tab</span>[<span class="pl-smi">final.tab</span><span class="pl-k">$</span><span class="pl-smi">name</span> <span class="pl-k">==</span> <span class="pl-smi">i</span>,c(<span class="pl-s"><span class="pl-pds">"</span>wilcox.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>deseq2.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mageck.fdr.neg<span class="pl-pds">"</span></span>)]
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Wilcox p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">1</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__DESeq2 p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">2</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__MAGeCK p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">3</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)  
      }
    <span class="pl-k">else</span>
      {
    <span class="pl-c"># Significantly?</span>
    <span class="pl-v">sig.pval.candidate</span> <span class="pl-k">=</span> <span class="pl-smi">final.tab</span>[<span class="pl-smi">final.tab</span><span class="pl-k">$</span><span class="pl-smi">name</span> <span class="pl-k">==</span> <span class="pl-smi">i</span>,c(<span class="pl-s"><span class="pl-pds">"</span>wilcox.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>deseq2.pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mageck.fdr.neg<span class="pl-pds">"</span></span>)]
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Wilcox p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">1</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__DESeq2 p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">2</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__MAGeCK p-value:__<span class="pl-pds">"</span></span>, <span class="pl-smi">sig.pval.candidate</span>[[<span class="pl-c1">3</span>]],<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
      }

  <span class="pl-c"># Scatterplot PAIRS GENES</span>
  <span class="pl-v">plothitsscatter.depleted</span> <span class="pl-k">=</span> carpools.hit.scatter(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">TREAT1.g</span>, <span class="pl-smi">TREAT2.g</span>, <span class="pl-smi">CONTROL1.g</span>, <span class="pl-smi">CONTROL2.g</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-smi">plot.labelcolor</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">median</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-smi">offsetplot</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.depleted</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>,  <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c">#cat("\\newpage","\n")</span>
    <span class="pl-c">#cat("\n")</span>
    <span class="pl-c"># Scatterplot PAIRS sgRNA</span>
    <span class="pl-v">plothitsscatter.depleted</span> <span class="pl-k">=</span> carpools.hit.scatter(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>, <span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>, <span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">title</span><span class="pl-k">=</span><span class="pl-smi">analysis.name</span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">labelcolor</span><span class="pl-k">=</span><span class="pl-smi">plot.labelcolor</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">normalize</span><span class="pl-k">=</span><span class="pl-smi">normalize</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">median</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-smi">offsetplot</span>, <span class="pl-v">center</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">aggregated</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.depleted</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>,  <span class="pl-v">pch</span><span class="pl-k">=</span><span class="pl-c1">16</span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c"># SgRNA foldchange, Z-Ratio, Vioplot</span>

    <span class="pl-c"># SgRNA foldchange, Z-Ratio, Vioplot</span>
    <span class="pl-v">sgrnas.dep</span> <span class="pl-k">=</span> carpools.hit.sgrna(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>, <span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">put.names</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>, <span class="pl-v">controls.target</span><span class="pl-k">=</span><span class="pl-smi">controls.target</span>, <span class="pl-v">controls.nontarget</span><span class="pl-k">=</span><span class="pl-smi">controls.nontarget</span>)

  par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))

    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\\</span>newpage<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    <span class="pl-c"># sgRNA table</span>
 <span class="pl-k">if</span>(<span class="pl-smi">count</span><span class="pl-k">==</span><span class="pl-c1">0</span>)
      {
      <span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">overlap.depleted</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, paste(<span class="pl-s"><span class="pl-pds">"</span>HITS-sgRNA<span class="pl-pds">"</span></span>, paste(<span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>List of Genes<span class="pl-pds">"</span></span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>, <span class="pl-v">col.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
      }

  <span class="pl-v">sgrnas.dep.table</span> <span class="pl-k">=</span> carpools.sgrna.table(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-k">list</span>(<span class="pl-smi">CONTROL1</span>, <span class="pl-smi">CONTROL2</span>, <span class="pl-smi">TREAT1</span>, <span class="pl-smi">TREAT2</span>), <span class="pl-v">dataset.names</span> <span class="pl-k">=</span> c(<span class="pl-smi">d.CONTROL1</span>, <span class="pl-smi">d.CONTROL2</span>, <span class="pl-smi">d.TREAT1</span>, <span class="pl-smi">d.TREAT2</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">fullmatchcolumn</span><span class="pl-k">=</span><span class="pl-smi">fullmatchcolumn</span>, <span class="pl-v">norm.function</span><span class="pl-k">=</span><span class="pl-smi">norm.function</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">labelgenes</span><span class="pl-k">=</span><span class="pl-smi">i</span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-smi">cutoff.override.depleted</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.hits.plot.enriched</span>, <span class="pl-v">sgrna.file</span> <span class="pl-k">=</span> <span class="pl-smi">libFILE</span>, <span class="pl-v">write</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

   <span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">sgrnas.dep.table</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, paste(<span class="pl-s"><span class="pl-pds">"</span>HITS-sgRNA<span class="pl-pds">"</span></span>, paste(<span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span>as.character(<span class="pl-smi">i</span>), <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">row.names</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span> )

print(<span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">sgrnas.dep.table</span>))

cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

<span class="pl-v">count</span> <span class="pl-k">=</span> <span class="pl-smi">count</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>
}
</pre></div>

<p>\newpage</p>

<h2><a id="user-content-compare-analysis" class="anchor" href="#compare-analysis" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Compare Analysis</h2>

<p>On the following pages, a comparison between the hit analysis methods is generated.  </p>

<h3><a id="user-content-enriched-4" class="anchor" href="#enriched-4" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enriched</h3>

<p><strong>List</strong>  </p>

<p>Moreover, the <strong><code>r number.hits.plot.enriched</code> % top enriched hits</strong> sorted according to <strong>MAGeCK</strong> are stored in <strong><code>r paste(analysis.name, "-COMPARE-HITS.xls", sep="")</code></strong> and are listed below:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.analysis.enriched</span> <span class="pl-k">=</span> compare.analysis(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.override</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.enriched.counter</span>, <span class="pl-v">output</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>list<span class="pl-pds">"</span></span>, <span class="pl-v">sort.by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>fdr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rank<span class="pl-pds">"</span></span>))

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">data.analysis.enriched</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>COMPARE-HITS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Enriched<span class="pl-pds">"</span></span>)
<span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.analysis.enriched</span>[,c(<span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">7</span>)])</pre></div>

<p>\newpage</p>

<p><strong>3D Scatterplot</strong><br>
This plot shows you the p-value of overlapping enriched genes of all three methods. The best hits will show up in the upper right corner.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#,fig.width=10,fig.height=10</span>
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))
<span class="pl-v">compared</span> <span class="pl-k">=</span> compare.analysis( <span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>enriched<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.enriched.counter</span>, <span class="pl-v">output</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>3dplot<span class="pl-pds">"</span></span>, <span class="pl-v">sort.by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rank<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rank<span class="pl-pds">"</span></span>), <span class="pl-v">plot.method</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>wilcox<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>deseq<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>), <span class="pl-v">plot.feature</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>fdr<span class="pl-pds">"</span></span>))
</pre></div>

<p>\newpage</p>

<h3><a id="user-content-depleted-3" class="anchor" href="#depleted-3" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Depleted</h3>

<p><strong>List</strong>  </p>

<p>Moreover, the <strong><code>r number.hits.plot.depleted</code> % top depleted hits</strong> sorted according to <strong>MAGeCK_ are stored in __<code>r paste(analysis.name, "-COMPARE-HITS.xls", sep="")</code></strong> and are listed below:  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-v">data.analysis.depleted</span> <span class="pl-k">=</span> compare.analysis(<span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.depleted.counter</span>, <span class="pl-v">output</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>list<span class="pl-pds">"</span></span>, <span class="pl-v">sort.by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>fdr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rank<span class="pl-pds">"</span></span>))

<span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">data.analysis.depleted</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>COMPARE-HITS.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Depleted<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

<span class="pl-e">knitr</span><span class="pl-k">::</span>kable(<span class="pl-smi">data.analysis.depleted</span>[,c(<span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">7</span>)])</pre></div>

<p>\newpage</p>

<p><strong>3D Scatterplot</strong>  </p>

<p>This plot shows you the p-value of overlapping depleted genes of all three methods. The best hits will show up in the upper right corner.</p>

<div class="highlight highlight-source-r"><pre>par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))
<span class="pl-v">compared2</span> <span class="pl-k">=</span> compare.analysis( <span class="pl-v">wilcox</span><span class="pl-k">=</span><span class="pl-smi">data.wilcox</span>, <span class="pl-v">deseq</span><span class="pl-k">=</span><span class="pl-smi">data.deseq</span>, <span class="pl-v">mageck</span><span class="pl-k">=</span><span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>depleted<span class="pl-pds">"</span></span>,  <span class="pl-v">cutoff.deseq</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">cutoff.wilcox</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">cutoff.mageck</span> <span class="pl-k">=</span> <span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">cutoff.override</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">cutoff.hits</span><span class="pl-k">=</span><span class="pl-smi">number.depleted.counter</span>, <span class="pl-v">output</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>3dplot<span class="pl-pds">"</span></span>, <span class="pl-v">sort.by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rank<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rank<span class="pl-pds">"</span></span>), <span class="pl-v">plot.method</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>wilcox<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>deseq<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>), <span class="pl-v">plot.feature</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>pval<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>fdr<span class="pl-pds">"</span></span>))
</pre></div>

<p>\newpage</p>

<h2><a id="user-content-visualize-analysis" class="anchor" href="#visualize-analysis" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Visualize Analysis</h2>

<p>The following pages present a visual representation of the p-value for single genes within all analysis methods.<br>
<em>Significantly</em> enriched genes, which had a corrected p-value below the threshold set for each method, are highlighted in <strong>red</strong>.<br>
<em>Significantly</em> depleted genes, which had a corrected p-value below the threshold set for each method, are highlighted in <strong>blue</strong>.  </p>

<p>Moreover, the number of significant sgRNAs is presented for MAGeCK as well.</p>

<h3><a id="user-content-wilcox-1" class="anchor" href="#wilcox-1" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Wilcox</h3>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#, fig.height=10, fig.width=15</span>
<span class="pl-v">wilcox.result</span> <span class="pl-k">=</span> carpools.hitident(<span class="pl-smi">data.wilcox</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>wilcox<span class="pl-pds">"</span></span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>wilcox<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">print.names</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-c1">1.2</span>, <span class="pl-v">plot.p</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.wilcox</span>, <span class="pl-v">separate</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
</pre></div>

<p>\newpage</p>

<h3><a id="user-content-deseq2-1" class="anchor" href="#deseq2-1" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>DESeq2</h3>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#, fig.height=10, fig.width=15</span>
<span class="pl-v">deseq.result</span> <span class="pl-k">=</span> carpools.hitident(<span class="pl-smi">data.deseq</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>deseq2<span class="pl-pds">"</span></span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>DESeq2<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">print.names</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">plot.p</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.deseq</span>, <span class="pl-v">separate</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-c1">1.2</span>, <span class="pl-v">sgRNA.top</span><span class="pl-k">=</span><span class="pl-smi">sgRNAs.top.deseq</span>)

</pre></div>

<p>\newpage</p>

<h3><a id="user-content-mageck-1" class="anchor" href="#mageck-1" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>MAGeCK</h3>

<div class="highlight highlight-source-r"><pre><span class="pl-c">#, fig.height=10, fig.width=15</span>

<span class="pl-v">mageck.result</span> <span class="pl-k">=</span> carpools.hitident(<span class="pl-smi">data.mageck</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>mageck<span class="pl-pds">"</span></span>, <span class="pl-v">title</span><span class="pl-k">=</span>paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>MAGeCK<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>), <span class="pl-v">inches</span><span class="pl-k">=</span><span class="pl-c1">0.1</span>, <span class="pl-v">print.names</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">separate</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">plot.p</span><span class="pl-k">=</span><span class="pl-smi">sig.pval.mageck</span>, <span class="pl-v">offsetplot</span><span class="pl-k">=</span><span class="pl-c1">1.2</span>, <span class="pl-v">sgRNA.top</span><span class="pl-k">=</span><span class="pl-smi">sgRNAs.top.mageck</span>)
</pre></div>

<p>\newpage</p>

<h1><a id="user-content-final-gene-table" class="anchor" href="#final-gene-table" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Final Gene Table</h1>

<p>A final table with all information for each gene is stored in<br>
<strong><code>r paste(datapath, paste(analysis.name, "FINAL.xls", sep="_"), sep="/")</code></strong>.</p>

<p>\newpage</p>

<h1><a id="user-content-annotate-hit-candidates" class="anchor" href="#annotate-hit-candidates" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Annotate Hit Candidates</h1>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(identical(<span class="pl-smi">g.convert</span>,<span class="pl-c1">TRUE</span>))
  {
    <span class="pl-v">g.identifier</span> <span class="pl-k">=</span> <span class="pl-smi">g.identifier.new</span> 
  }
<span class="pl-k">if</span>(identical(<span class="pl-smi">a.annotate.hits</span>, <span class="pl-c1">TRUE</span>))
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> Hit candidates are annotated with additional information from __biomaRt__. <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> The following information is retrieved: <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Database:__<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>, <span class="pl-smi">a.database</span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Dataset:__<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>, <span class="pl-smi">a.dataset</span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>__Filters:__<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span>, <span class="pl-smi">a.annotate</span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    cat(<span class="pl-s"><span class="pl-pds">"</span>The annotated list of all hit candidates is stored in<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>, <span class="pl-smi">datapath</span>,<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
    cat(<span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>ANNOTATION.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>),<span class="pl-s"><span class="pl-pds">"</span>**<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)

    <span class="pl-v">annotate.enriched.all</span> <span class="pl-k">=</span> get.gene.info(as.data.frame(<span class="pl-smi">overlap.enriched</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> <span class="pl-smi">a.annotate</span>, <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>info<span class="pl-pds">"</span></span>, <span class="pl-v">controls</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)  

  <span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">annotate.enriched.all</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>ANNOTATION.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Enriched<span class="pl-pds">"</span></span>)

  <span class="pl-v">annotate.depleted.all</span> <span class="pl-k">=</span> get.gene.info(as.data.frame(<span class="pl-smi">overlap.depleted</span>), <span class="pl-v">namecolumn</span><span class="pl-k">=</span><span class="pl-smi">namecolumn</span>, <span class="pl-v">extractpattern</span><span class="pl-k">=</span><span class="pl-smi">g.extractpattern</span>, <span class="pl-v">database</span><span class="pl-k">=</span><span class="pl-smi">a.database</span>, <span class="pl-v">dataset</span><span class="pl-k">=</span><span class="pl-smi">a.dataset</span>, <span class="pl-v">filters</span><span class="pl-k">=</span><span class="pl-smi">g.identifier</span>, <span class="pl-v">attributes</span> <span class="pl-k">=</span> <span class="pl-smi">a.annotate</span>, <span class="pl-k">return</span><span class="pl-v">.val</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>info<span class="pl-pds">"</span></span>, <span class="pl-v">controls</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

  <span class="pl-e">xlsx</span><span class="pl-k">::</span>write.xlsx(<span class="pl-smi">annotate.depleted.all</span>, <span class="pl-v">file</span><span class="pl-k">=</span>paste(<span class="pl-smi">datapath</span>, paste(<span class="pl-smi">analysis.name</span>, <span class="pl-s"><span class="pl-pds">"</span>ANNOTATION.xls<span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>), <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>/<span class="pl-pds">"</span></span>), <span class="pl-v">sheetName</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Depleted<span class="pl-pds">"</span></span>, <span class="pl-v">append</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)

  } <span class="pl-k">else</span>
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span> Hit candidates are __not annotated__ with additional information from __biomaRt__. <span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
  }
</pre></div>

<h1><a id="user-content-data-extraction-mapping-and-files" class="anchor" href="#data-extraction-mapping-and-files" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Data Extraction, Mapping and Files</h1>

<table><thead>
<tr>
<th>Dataset</th>
<th>.fastq file name</th>
<th>Description</th>
</tr>
</thead><tbody>
<tr>
<td>Control #1</td>
<td><code>r fileCONTROL1</code></td>
<td><code>r d.CONTROL1</code></td>
</tr>
<tr>
<td>Control #2</td>
<td><code>r fileCONTROL2</code></td>
<td><code>r d.CONTROL2</code></td>
</tr>
<tr>
<td>Treatment #1</td>
<td><code>r fileTREAT1</code></td>
<td><code>r d.TREAT1</code></td>
</tr>
<tr>
<td>Treatment #2</td>
<td><code>r fileTREAT2</code></td>
<td><code>r d.TREAT2</code></td>
</tr>
</tbody></table>

<p>The data is located in<br>
<em><code>r datapath</code></em>  </p>

<p>and the script files for data extraction and mapping are located in<br>
<em><code>r scriptpath</code></em>.  </p>

<p><strong>All file-based output (e.g. tables) from MAGeCK is stored in:</strong><br>
<em><code>r getwd()</code></em>.  </p>

<div class="highlight highlight-source-r"><pre><span class="pl-k">if</span>(<span class="pl-smi">extract</span>)
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>__FASTQ files were extracted.__<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
  } 

<span class="pl-k">if</span>(<span class="pl-smi">mapping</span>) 
  {
    cat(<span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>FASTQ files were mapped using <span class="pl-pds">"</span></span>, <span class="pl-smi">referencefile</span>, <span class="pl-s"><span class="pl-pds">"</span>.fasta as a reference. <span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
  }
</pre></div>

<table><thead>
<tr>
<th>Parameter</th>
<th>Value</th>
</tr>
</thead><tbody>
<tr>
<td>Reverse Complement Sequence</td>
<td><code>r reversecomplement</code></td>
</tr>
<tr>
<td>Pattern of Data Extraction</td>
<td><code>r seq.pattern</code></td>
</tr>
<tr>
<td>Maschine Identifier FASTq</td>
<td><code>r maschine.pattern</code></td>
</tr>
<tr>
<td>Create Bowtie2 Index?</td>
<td><code>r createindex</code></td>
</tr>
<tr>
<td>Reference .fasta File</td>
<td><code>r referencefile</code></td>
</tr>
<tr>
<td>Bt2 Threads</td>
<td><code>r threads</code></td>
</tr>
<tr>
<td>Bt2 Sensitivity</td>
<td><code>r sensitivity</code></td>
</tr>
<tr>
<td>sgRNA Oligo Match</td>
<td><code>r match</code></td>
</tr>
</tbody></table>
</article>
  </div>

</div>

<button type="button" data-facebox="#jump-to-line" data-facebox-class="linejump" data-hotkey="l" class="hidden">Jump to Line</button>
<div id="jump-to-line" style="display:none">
  <!-- </textarea> --><!-- '"` --><form accept-charset="UTF-8" action="" class="js-jump-to-line-form" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
    <input class="form-control linejump-input js-jump-to-line-field" type="text" placeholder="Jump to line&hellip;" aria-label="Jump to line" autofocus>
    <button type="submit" class="btn">Go</button>
</form></div>

  </div>
  <div class="modal-backdrop"></div>
</div>


    </div>
  </div>

    </div>

        <div class="container site-footer-container">
  <div class="site-footer" role="contentinfo">
    <ul class="site-footer-links right">
        <li><a href="https://status.github.com/" data-ga-click="Footer, go to status, text:status">Status</a></li>
      <li><a href="https://developer.github.com" data-ga-click="Footer, go to api, text:api">API</a></li>
      <li><a href="https://training.github.com" data-ga-click="Footer, go to training, text:training">Training</a></li>
      <li><a href="https://shop.github.com" data-ga-click="Footer, go to shop, text:shop">Shop</a></li>
        <li><a href="https://github.com/blog" data-ga-click="Footer, go to blog, text:blog">Blog</a></li>
        <li><a href="https://github.com/about" data-ga-click="Footer, go to about, text:about">About</a></li>

    </ul>

    <a href="https://github.com" aria-label="Homepage" class="site-footer-mark" title="GitHub">
      <svg aria-hidden="true" class="octicon octicon-mark-github" height="24" version="1.1" viewBox="0 0 16 16" width="24"><path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path></svg>
</a>
    <ul class="site-footer-links">
      <li>&copy; 2016 <span title="0.08820s from github-fe151-cp1-prd.iad.github.net">GitHub</span>, Inc.</li>
        <li><a href="https://github.com/site/terms" data-ga-click="Footer, go to terms, text:terms">Terms</a></li>
        <li><a href="https://github.com/site/privacy" data-ga-click="Footer, go to privacy, text:privacy">Privacy</a></li>
        <li><a href="https://github.com/security" data-ga-click="Footer, go to security, text:security">Security</a></li>
        <li><a href="https://github.com/contact" data-ga-click="Footer, go to contact, text:contact">Contact</a></li>
        <li><a href="https://help.github.com" data-ga-click="Footer, go to help, text:help">Help</a></li>
    </ul>
  </div>
</div>



    

    <div id="ajax-error-message" class="ajax-error-message flash flash-error">
      <svg aria-hidden="true" class="octicon octicon-alert" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"></path></svg>
      <button type="button" class="flash-close js-flash-close js-ajax-error-dismiss" aria-label="Dismiss error">
        <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
      </button>
      Something went wrong with that request. Please try again.
    </div>


      
      <script crossorigin="anonymous" integrity="sha256-cWK+6icqhW0G4ISUX9gCake7uefrKV2Vtg6oKybCcpY=" src="https://assets-cdn.github.com/assets/frameworks-7162beea272a856d06e084945fd8026a47bbb9e7eb295d95b60ea82b26c27296.js"></script>
      <script async="async" crossorigin="anonymous" integrity="sha256-y2q3K8g+kOg6QpsET0YFctxepvsbWJT8GFx56TiZhvA=" src="https://assets-cdn.github.com/assets/github-cb6ab72bc83e90e83a429b044f460572dc5ea6fb1b5894fc185c79e9389986f0.js"></script>
      
      
      
      
      
      
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-banner hidden">
      <svg aria-hidden="true" class="octicon octicon-alert" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"></path></svg>
      <span class="signed-in-tab-flash">You signed in with another tab or window. <a href="">Reload</a> to refresh your session.</span>
      <span class="signed-out-tab-flash">You signed out in another tab or window. <a href="">Reload</a> to refresh your session.</span>
    </div>
    <div class="facebox" id="facebox" style="display:none;">
  <div class="facebox-popup">
    <div class="facebox-content" role="dialog" aria-labelledby="facebox-header" aria-describedby="facebox-description">
    </div>
    <button type="button" class="facebox-close js-facebox-close" aria-label="Close modal">
      <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"></path></svg>
    </button>
  </div>
</div>

  </body>
</html>

